

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>GOMC 2.40 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html#document-index" class="icon icon-home"> GOMC
          

          
          </a>

          
            
            
          

          

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
            
              <!-- Local TOC -->
              <div class="local-toc"><p class="caption"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="index.html#document-overview">Overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="index.html#document-introduction">Introduction</a><ul>
<li class="toctree-l2"><a class="reference internal" href="index.html#gomc-supported-ensembles">GOMC supported ensembles:</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#gomc-supported-monte-carlo-moves">GOMC supported Monte Carlo moves:</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#gomc-supported-force-fields">GOMC supported force fields:</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#gomc-supported-molecules">GOMC supported molecules:</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="index.html#document-software_requirements">Software Requirements</a><ul>
<li class="toctree-l2"><a class="reference internal" href="index.html#c-11-compliant-compiler">C++11 Compliant Compiler</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#cmake">CMake</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#cuda-toolkit">CUDA Toolkit</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="index.html#document-software_tools">Recommended Software Tools</a><ul>
<li class="toctree-l2"><a class="reference internal" href="index.html#packmol">Packmol</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#vmd">VMD</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="index.html#document-download">How to get the software</a><ul>
<li class="toctree-l2"><a class="reference internal" href="index.html#github">GitHub</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#website">Website</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="index.html#document-compiling">Compiling GOMC</a><ul>
<li class="toctree-l2"><a class="reference internal" href="index.html#linux">Linux</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#windows">Windows</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#configuring-cmake">Configuring CMake</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="index.html#document-input_file">Input File Formats</a><ul>
<li class="toctree-l2"><a class="reference internal" href="index.html#pdb-file">PDB File</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#psf-file">PSF File</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#topology-file">Topology File</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#parameter-file-s">Parameter File(s)</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#exotic-or-mie-parameter-file">Exotic or Mie Parameter File</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#control-file-conf">Control File (*.conf)</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="index.html#document-output_file">GOMC’s Output Files, Terminal Output</a><ul>
<li class="toctree-l2"><a class="reference internal" href="index.html#console-output">Console Output</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#block-output-files">Block Output Files</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#visualizing-simulation">Visualizing Simulation</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="index.html#document-putting_all_together">Putting it all together: Running a GOMC Simulation</a></li>
<li class="toctree-l1"><a class="reference internal" href="index.html#document-vdw_energy">Intermolecular Energy and Virial Function (Van der Waals)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="index.html#vdw">VDW</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#shift">SHIFT</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#switch">SWITCH</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#switch-martini">SWITCH (MARTINI)</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="index.html#document-electrostatic">Intermolecular Energy and Virial Function (Electrostatic)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="index.html#ewald">Ewald</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#shift">SHIFT</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#switch">SWITCH</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#switch-martini">SWITCH (MARTINI)</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="index.html#document-help">Get Help or Technical Support</a></li>
</ul>
</div>
            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html#document-index">GOMC</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html#document-index">Docs</a> &raquo;</li>
        
      <li>GOMC 2.40 documentation</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="welcome-to-gomc-s-documentation">
<h1>Welcome to GOMC’s documentation!<a class="headerlink" href="#welcome-to-gomc-s-documentation" title="Permalink to this headline">¶</a></h1>
<div class="toctree-wrapper compound">
<span id="document-overview"></span><div class="section" id="overview">
<h2>Overview<a class="headerlink" href="#overview" title="Permalink to this headline">¶</a></h2>
<p>This document will instruct a new user how to download, compile, prepare the input files, and run the GOMC molecular simulation code. A basic understanding of statistical physics is recommended to complete this tutorial.</p>
<p>To demonstrate the capabilities of the code, the user is guided through the process of downloading, compiling a GOMC executable, and preparing input files such as PDB, PSF, Parameter, and Configuration file. Executable is then used to calculate the saturated vapor and liquid equilibria (VLE) using Gibbs Ensemble Monte Carlo on systems of pure isobutane (R600a), a branched alkane that whose application as a refrigerant/propellant is increasing. The Transferable Potentials for Phase Equilibria (TraPPE) united atom (UA) force field is used to describe the molecular geometry constraints and the intermolecular interactions.</p>
<p><a class="reference external" href="http://en.wikipedia.org/wiki/Isobutane">http://en.wikipedia.org/wiki/Isobutane</a></p>
</div>
<span id="document-introduction"></span><div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>GPU Optimized Monte Carlo (GOMC) is open-source software for simulating many-body molecular systems using the Metropolis Monte Carlo algorithm. GOMC is written in object oriented C++, which was chosen since it offers a good balance between code development time, interoperability with existing software elements, and code performance. The software may be compiled as a single-threaded application, a multi-threaded application using OpenMP, or to use many-core heterogeneous CPU-GPU architectures using OpenMP and CUDA. GOMC officially supports Windows 7 or newer and most modern distribution of GNU/Linux. This software has the ability to compile on recent versions of macOS; however, such a platform is not officially supported.</p>
<p>GOMC employs widely-used simulation file types (PDB, PSF, CHARMM-style parameter file) and supports polar and non-polar linear and branched molecules. GOMC can be used to study vapor-liquid and liquid-liquid equilibria, adsorption in porous materials, surfactant self-assembly, and condensed phase structure for complex molecules.</p>
<p>To cite GOMC software, please refer to <a class="reference external" href="https://www.sciencedirect.com/science/article/pii/S2352711018301171">GOMC paper</a>.</p>
<div class="section" id="gomc-supported-ensembles">
<h3>GOMC supported ensembles:<a class="headerlink" href="#gomc-supported-ensembles" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li>Canonical (NVT)</li>
<li>Isobaric-isothermal (NPT)</li>
<li>Grand canonical (μVT)</li>
<li>Constant volume Gibbs (NVT-Gibbs)</li>
<li>Constant pressure Gibbs (NPT-Gibbs)</li>
</ul>
</div>
<div class="section" id="gomc-supported-monte-carlo-moves">
<h3>GOMC supported Monte Carlo moves:<a class="headerlink" href="#gomc-supported-monte-carlo-moves" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li>Rigid-body displacement</li>
<li>Rigid-body rotation</li>
<li>Regrowth using <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/jp984742e">coupled-decoupled configurational-bias</a></li>
<li>Crankshaft using combination of <a class="reference external" href="https://aip.scitation.org/doi/abs/10.1063/1.438608">crankshaft</a> and <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/jp984742e">coupled-decoupled configurational-bias</a></li>
<li>Intra-box swap using <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/jp984742e">coupled-decoupled configurational-bias</a></li>
<li>Intra-box <a class="reference external" href="https://aip.scitation.org/doi/abs/10.1063/1.5025184">molecular exchange Monte Carlo</a></li>
<li>Inter-box swap using <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/jp984742e">coupled-decoupled configurational-bias</a></li>
<li>Inter-box <a class="reference external" href="https://www.sciencedirect.com/science/article/pii/S0378381218305351">molecular exchange monter carlo</a></li>
<li>Volume exchange (both isotropic and anisotropic)</li>
</ul>
</div>
<div class="section" id="gomc-supported-force-fields">
<h3>GOMC supported force fields:<a class="headerlink" href="#gomc-supported-force-fields" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li>OPLS</li>
<li>CHARMM</li>
<li>TraPPE</li>
<li>Mie</li>
<li>Martini</li>
</ul>
</div>
<div class="section" id="gomc-supported-molecules">
<h3>GOMC supported molecules:<a class="headerlink" href="#gomc-supported-molecules" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li>Polar molecules (using Ewald summation)</li>
<li>Non-polar molecules (standard LJ and Mie potential)</li>
<li>Linear molecules (using <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/jp984742e">coupled-decoupled configurational-bias</a>)</li>
<li>Branched molecules (using <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/jp984742e">coupled-decoupled configurational-bias</a>)</li>
<li>Cyclic molecules (using combination of <a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/jp984742e">coupled-decoupled configurational-bias</a> and <a class="reference external" href="https://aip.scitation.org/doi/abs/10.1063/1.3644939">crankshaft</a> to sample intramolecular degrees of freedom of cyclic molecules)</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>It is important to start the simulation with correct molecular geometry such as correct bond length, angles, and dihedral.</li>
<li>In GOMC if the defined bond length in <code class="docutils literal notranslate"><span class="pre">Parameter</span></code> file is different from calculated bond length in <code class="docutils literal notranslate"><span class="pre">PDB</span></code> files by more than 0.02 <span class="math notranslate nohighlight">\(Å\)</span>, you will receive a warning message with detailed information (box, residue id, specified bond length, and calculated bond length)</li>
</ul>
</div>
<div class="admonition important">
<p class="first admonition-title">Important</p>
<ul class="last simple">
<li>Molecular geometry of <code class="docutils literal notranslate"><span class="pre">Linear</span></code> and <code class="docutils literal notranslate"><span class="pre">Branched</span></code> molecules will be corrected during the simulation by using the Monte Carlo moves that uses coupled-decoupled configurational-bias method, such as <code class="docutils literal notranslate"><span class="pre">Regrowth</span></code>, <code class="docutils literal notranslate"><span class="pre">Intra-box</span> <span class="pre">swap</span></code>, and <code class="docutils literal notranslate"><span class="pre">Inter-box</span> <span class="pre">swap</span></code>.</li>
</ul>
</div>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<ul class="last simple">
<li>Bond length of the <code class="docutils literal notranslate"><span class="pre">Cyclic</span></code> molecules that belong to the body of rings will never be changed. Incorrect bond length may result in incorrect simulation results.</li>
<li>To sample the angles and dihedrals of a <code class="docutils literal notranslate"><span class="pre">Cyclic</span></code> molecule that belongs to the body of the ring, <code class="docutils literal notranslate"><span class="pre">Regrowth</span></code> or <code class="docutils literal notranslate"><span class="pre">Crankshaft</span></code> Monte Carlo move must be used.</li>
<li>Any atom or group attached to the body of the ring, will uses coupled-decoupled configurational-bias to sample the molecular geometry.</li>
<li>Flexible <code class="docutils literal notranslate"><span class="pre">Cyclic</span></code> molecules with multiple rings (3 or more) that share edges (e.g. tricyclic), are not supported in GOMC. This is due the fact that no <code class="docutils literal notranslate"><span class="pre">Crankshaft</span></code> move can alter the angle or dihedral of this atom, without changing the bond length.</li>
</ul>
</div>
</div>
</div>
<span id="document-software_requirements"></span><div class="section" id="software-requirements">
<h2>Software Requirements<a class="headerlink" href="#software-requirements" title="Permalink to this headline">¶</a></h2>
<div class="section" id="c-11-compliant-compiler">
<h3>C++11 Compliant Compiler<a class="headerlink" href="#c-11-compliant-compiler" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">Linux/macOS</p>
<ul>
<li><p class="first">icpc (Intel C++ Compiler)</p>
<p>In Linux, the Intel compiler will generally produce the fastest CPU executables (when running on Intel Core processors). Type the following command in a terminal:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ icpc --version
</pre></div>
</div>
<p>If gives a version number 16.0.3 (2016 Initial version) or later, you’re all set. Otherwise, we recommend upgrading.</p>
</li>
<li><p class="first">g++</p>
<p>Type the following command in a terminal:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ g++ --version
</pre></div>
</div>
<p>If gives a version number 4.4 or later, you’re all set. Otherwise, we recommend upgrading.</p>
</li>
</ul>
</li>
<li><p class="first">Windows</p>
<p>Visual Studio Microsoft’s Visual Studio 2010 or later is recommended.</p>
<p>To check the version:</p>
<p><em>Help</em> (top tab) -&gt; <em>About Microsoft Visual Studio</em></p>
<img alt="_images/vshelp.png" src="_images/vshelp.png" />
<img alt="_images/vsabout.png" src="_images/vsabout.png" />
</li>
</ul>
</div>
<div class="section" id="cmake">
<h3>CMake<a class="headerlink" href="#cmake" title="Permalink to this headline">¶</a></h3>
<p>To check if cmake is installed:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ which cmake
</pre></div>
</div>
<p>To check the version number:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ cmake --version
</pre></div>
</div>
<p>The minimum required version is 2.8. However, we recommend to use version 3.2 or later.</p>
</div>
<div class="section" id="cuda-toolkit">
<h3>CUDA Toolkit<a class="headerlink" href="#cuda-toolkit" title="Permalink to this headline">¶</a></h3>
<p>CUDA is required to compile the GPU executable in both Windows and Linux. However, is not required to compile the CPU code. To download and install CUDA visit NVIDIA’s webpage:</p>
<p><a class="reference external" href="https://developer.nvidia.com/cuda-downloads">https://developer.nvidia.com/cuda-downloads</a></p>
<p><a class="reference external" href="https://developer.nvidia.com/cuda">https://developer.nvidia.com/cuda</a></p>
<p>Please refer to CUDA Developer webpages to select an appropriate version for the desired platform. To install CUDA in Linux root/sudo, privileges are generally required. In Windows, administrative access is required.</p>
<p>To check if nvcc is installed:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ which nvcc
</pre></div>
</div>
<p>To check the version number:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ nvcc --version
</pre></div>
</div>
<p>The GPU builds of the code requires NVIDIA’s CUDA 8.0 or newer.</p>
</div>
</div>
<span id="document-software_tools"></span><div class="section" id="recommended-software-tools">
<h2>Recommended Software Tools<a class="headerlink" href="#recommended-software-tools" title="Permalink to this headline">¶</a></h2>
<p>The listed programs are used in this manual and are generally considered necessary.</p>
<div class="section" id="packmol">
<h3>Packmol<a class="headerlink" href="#packmol" title="Permalink to this headline">¶</a></h3>
<p>Packmol is a free molecule packing tool (written in Fortran), created by José Mario Martínez, a professor of mathematics at the State University of Campinas, Brazil. Packmol allows a specified number of molecules to be packed at defined separating distances within a certain region of space. More information regarding downloading and installing Packmol is available on their homepage:</p>
<p><a class="reference external" href="http://www.ime.unicamp.br/~martinez/packmol">http://www.ime.unicamp.br/~martinez/packmol</a></p>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">One of Packmol’s limitations is that it is unaware of topology; it treats each molecule or group of molecules as a rigid set of points. It is highly suggested to used the optimized structure of the molecule as the input file to packmol.</p>
</div>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">Another more serious limitation is that it is not aware of periodic boundary conditions (PBC). As a result, when using Packmol to pack PDBs for GOMC, it is recommended to pack to a box 1 Angstroms smaller than the simulation box size. This prevents hard overlaps over the periodic boundary.</p>
</div>
</div>
<div class="section" id="vmd">
<h3>VMD<a class="headerlink" href="#vmd" title="Permalink to this headline">¶</a></h3>
<p>VMD (Visual Molecular Dynamics) is a 3-D visualization and manipulation engine for molecular systems written in C-language. VMD is distributed and maintained by the University of Illinois at Urbana-Champaign. Its sources and binaries are free to download. It comes with a robust scripting engine, which is capable of running python and tcl scripts. More info can be found here:</p>
<p><a class="reference external" href="http://www.ks.uiuc.edu/Research/vmd/">http://www.ks.uiuc.edu/Research/vmd/</a></p>
<p>Although GOMC uses the same fundamental file types, PDB (coordinates) and PSF (topology) as VMD, it uses some special tricks to obey certain rules of those file formats. One useful purpose of VMD is visualization and analyze your systems.</p>
<div class="figure" id="id1">
<img alt="_images/vmd.png" src="_images/vmd.png" />
<p class="caption"><span class="caption-text">A system of united atom isobutane molecules</span></p>
</div>
<p>Nonetheless, the most critical part of VMD is a tool called PSFGen. PSFGen uses a tcl or python script to generate a PDB and PSF file for a system of one or more molecules. It is, perhaps, the most convenient way to generate a compliant PSF file.</p>
<div class="figure" id="id2">
<img alt="_images/psfgen.png" src="_images/psfgen.png" />
<p class="caption"><span class="caption-text">An overview of the PSFGen file generation process and its relationship to VMD/NAMD</span></p>
</div>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p>To read more about PSFGen, reference:</p>
<p><a class="reference external" href="http://www.ks.uiuc.edu/Research/vmd/plugins/psfgen">Plugin homepage &#64; UIUC</a></p>
<p><a class="reference external" href="http://www.ks.uiuc.edu/Training/Tutorials/namd/namd-tutorial-html/node6.html">Generating a Protein Structure File (PSF), part of the NAMD Tutorial from UIUC</a></p>
<p class="last"><a class="reference external" href="http://www.ks.uiuc.edu/Research/vmd/plugins/psfgen/ug.pdf">In-Depth Overview [PDF]</a></p>
</div>
</div>
</div>
<span id="document-download"></span><div class="section" id="how-to-get-the-software">
<h2>How to get the software<a class="headerlink" href="#how-to-get-the-software" title="Permalink to this headline">¶</a></h2>
<p>The CPU and GPU code are merged together under GOMC project. Currently, version control is handled through the GitHub repository. The latest GOMC release, Example files, and User Manual can be downloaded from GOMC website or GitHub repository.</p>
<div class="section" id="github">
<h3>GitHub<a class="headerlink" href="#github" title="Permalink to this headline">¶</a></h3>
<p>The posted builds in Master branch are “frozen” versions of the code that have been validated for a number of systems and ensembles. Other branches are created as a means of implementing new features. The latest updated code builds, manual, example files, and other resources can be obtained via the following GitHub repository:</p>
<p><a class="reference external" href="https://github.com/GOMC-WSU">GOMC GitHub Repository</a></p>
<img alt="_images/github.png" src="_images/github.png" />
<p>GOMC and Examples repository can be found under the main page. Under GOMC repository, the code and manual can be found. Each repository can be downloaded by clicking on the Clone or download tab.</p>
<img alt="_images/clone.png" src="_images/clone.png" />
<p>To clone the GOMC using git, execute the following command in your terminal:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ git clone https://github.com/GOMC-WSU/GOMC.git
</pre></div>
</div>
<p>To clone the GOMC Example files using git, execute the following command in your terminal:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ git clone https://github.com/GOMC-WSU/GOMC_Examples.git
</pre></div>
</div>
</div>
<div class="section" id="website">
<h3>Website<a class="headerlink" href="#website" title="Permalink to this headline">¶</a></h3>
<p>To access the GOMC website, please click on the following link:
<a class="reference external" href="http://gomc.eng.wayne.edu/">GOMC Website</a></p>
<p>The code can be found under the download tab, below and to the right of the logo. When new betas (or release builds) are announced, they will replace the prior code under the downloads tab. An announcement will be posted on the front page to notify users.</p>
<img alt="_images/website.jpg" src="_images/website.jpg" />
<p>GOMC is distributed as a compressed folder, containing the source and build system. To compile the code after downloading it, the first step is to extract the compressed build folder.</p>
<p>In Linux, the GPU and CPU codes are compressed using gzip and tar (*.tar.gz). To extract, simply move to the desire folder and type in the command line:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ tar -xzvf &lt;file name&gt;.tar.gz
</pre></div>
</div>
</div>
</div>
<span id="document-compiling"></span><div class="section" id="compiling-gomc">
<h2>Compiling GOMC<a class="headerlink" href="#compiling-gomc" title="Permalink to this headline">¶</a></h2>
<p>GOMC generates four executable files for CPU code; <code class="docutils literal notranslate"><span class="pre">GOMC_CPU_GEMC</span></code> (Gibbs ensemble), <code class="docutils literal notranslate"><span class="pre">GOMC_CPU_NVT</span></code> (NVT ensemble), <code class="docutils literal notranslate"><span class="pre">GOMC_CPU_NPT</span></code> (isobaric-isothermal ensemble), and <code class="docutils literal notranslate"><span class="pre">GOMC_CPU_GCMC</span></code> (Grand canonical ensemble). In case of installing CUDA Toolkit, GOMC will generate additional four executable files for GPU code; <code class="docutils literal notranslate"><span class="pre">GOMC_GPU_GEMC</span></code>, <code class="docutils literal notranslate"><span class="pre">GOMC_GPU_NVT</span></code>, <code class="docutils literal notranslate"><span class="pre">GOMC_GPU</span> <span class="pre">NPT</span></code>, and <code class="docutils literal notranslate"><span class="pre">GOMC_GPU_GCMC</span></code>.</p>
<p>This section guid users to compile GOMC in Linux or Windows.</p>
<div class="section" id="linux">
<h3>Linux<a class="headerlink" href="#linux" title="Permalink to this headline">¶</a></h3>
<p>First, navigate your command line to the GOMC base directory. To compile GOMC on Linux, give permission to “metamake.sh” by running the following command and execute it:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ chmod u+x metamake.sh
$ ./metamake.sh
</pre></div>
</div>
<p>This script will create a bin directory and run cmake file to compile the code as well. All executable files will be generated in the “bin” directory.</p>
</div>
<div class="section" id="windows">
<h3>Windows<a class="headerlink" href="#windows" title="Permalink to this headline">¶</a></h3>
<p>To compile GOMC on in Windows, follow these steps:</p>
<ol class="arabic simple">
<li>Open the Windows-compatible CMake GUI.</li>
<li>Set the Source Folder to the GOMC root folder.</li>
<li>Set the build Folder to your Build Folder.</li>
<li>Click configure, select your compiler/environment</li>
<li>Wait for CMake to finish the configuration.</li>
<li>Click configure again and click generate.</li>
<li>Open the CMake-generated project/solution etc. to the desired IDE (e.g Visual Studio).</li>
<li>Using the solution in the IDE of choice build GOMC per the IDE’s standard release compilation/exe- cutable generation methods.</li>
</ol>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">You can also use CMake from the Windows command line if its directory is added to the PATH environment variable.</p>
</div>
</div>
<div class="section" id="configuring-cmake">
<h3>Configuring CMake<a class="headerlink" href="#configuring-cmake" title="Permalink to this headline">¶</a></h3>
<p>GOMC uses CMAKE to generate multi-platform intermediate files to compile the project. In this section, you can find all the information needed to configure CMake.
We recommend using a different directory for the CMake output than the home directory of the project as CMake tend to generate lots of files.
CMake has a ridiculously expansive set of options, so this document will only reproduce the most obviously relevant ones. When possible, options should be passed into CMake via command line options rather than the CMakeCached.txt file:</p>
<dl class="docutils">
<dt>CMAKE_BUILD_TYPE</dt>
<dd><p class="first">To get the best performance you should build the project in release mode. In CMake GUI you can set the value of “CMAKE_BUILD_TYPE” to “Release” and in CMake command line you can add the following to the CMake:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>-DCMAKE_BUILD_TYPE<span class="o">=</span>Release
</pre></div>
</div>
<p>To compile the GOMC in debug mode, in CMake GUI, change the value of “CMAKE_BUILD_TYPE” to “Debug” and in CMake command line you can add the following to the CMake:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>-DCMAKE_BUILD_TYPE<span class="o">=</span>Debug
</pre></div>
</div>
<p class="last">Other options are “&lt;None | ReleaseWithDebInfo | MinSizeRel&gt;”.</p>
</dd>
<dt>CMAKE_CXX_COMPILER</dt>
<dd>This option will set the compiler. It is recommended to use the Intel Compiler and linking tools, if possible (icc/icpc/etc.). They significantly outperform the default GNU and Visual Studio compiler tools and are available for free for academic use with registration.</dd>
<dt>CMAKE_CXX_FLAGS_RELEASE:STRING</dt>
<dd><p class="first">To run the parallel version of CPU code, it needs to be compiled with openmp library. Open the file “CMakeCache.txt”, while still in the “bin” folder, and change the value from “-O3 -DNDEBUG” to “-O3 -qopenmp -DNDEBUG”. Recompile the GOMC by typing the command:</p>
<div class="last highlight-bash notranslate"><div class="highlight"><pre><span></span>$ make
</pre></div>
</div>
</dd>
<dt>ENSEMBLE_NVT</dt>
<dd>You can turn the compilation of CPU version of NVT ensemble on or off using this option.
-DENSEMBLE_NVT=&lt;On | Off&gt;</dd>
<dt>ENSEMBLE_NPT</dt>
<dd>You can turn the compilation of CPU version of NPT ensemble on or off using this option.
-DENSEMBLE_NPT=&lt;On | Off&gt;</dd>
<dt>ENSEMBLE_GCMC</dt>
<dd>You can turn the compilation of CPU version of GCMC ensemble on or off using this option.
-DENSEMBLE_GCMC=&lt;On | Off&gt;</dd>
<dt>ENSEMBLE_GEMC</dt>
<dd>You can turn the compilation of CPU version of GEMC ensemble on or off using this option.
-DENSEMBLE_GEMC=&lt;On | Off&gt;</dd>
<dt>ENSEMBLE_GPU_NVT</dt>
<dd>You can turn the compilation of GPU version of NVT ensemble on or off using this option.
-DENSEMBLE_NVT=&lt;On | Off&gt;</dd>
<dt>ENSEMBLE_GPU_NPT</dt>
<dd>You can turn the compilation of GPU version of NPT ensemble on or off using this option.
-DENSEMBLE_NPT=&lt;On | Off&gt;</dd>
<dt>ENSEMBLE_GPU_GCMC</dt>
<dd>You can turn the compilation of GPU version of GCMC ensemble on or off using this option.
-DENSEMBLE_GCMC=&lt;On | Off&gt;</dd>
<dt>ENSEMBLE_GPU_GEMC</dt>
<dd>You can turn the compilation of GPU version of GEMC ensemble on or off using this option.
-DENSEMBLE_GEMC=&lt;On | Off&gt;</dd>
</dl>
</div>
</div>
<span id="document-input_file"></span><div class="section" id="input-file-formats">
<h2>Input File Formats<a class="headerlink" href="#input-file-formats" title="Permalink to this headline">¶</a></h2>
<p>In order to run simulation in GOMC, the following files need to be provided:</p>
<ul class="simple">
<li>GOMC executable</li>
<li>PDB file(s)</li>
<li>PSF file(s)</li>
<li>Parameter file</li>
<li>Input file “NAME.conf” (proprietary control file)</li>
</ul>
<div class="section" id="pdb-file">
<h3>PDB File<a class="headerlink" href="#pdb-file" title="Permalink to this headline">¶</a></h3>
<p>GOMC requires only one PDB file for NVT and NPT ensembles. However, GOMC requires two PDB files for GEMC and GCMC ensembles.</p>
<div class="section" id="what-is-pdb-file">
<h4>What is PDB file<a class="headerlink" href="#what-is-pdb-file" title="Permalink to this headline">¶</a></h4>
<p>The term PDB can refer to the Protein Data Bank (<a class="reference external" href="http://www.rcsb.org/pdb/">http://www.rcsb.org/pdb/</a>), to a data file provided there, or to any file following the PDB format. Files in the PDB include various information such as the name of the compound, the ATOM and HETATM records containing the coordinates of the molecules, and etc. PDB widely used by NAMD, GROMACS, CHARMM, ACEMD, and Amber. GOMC ignore everything in a PDB file except for the REMARK, CRYST1, ATOM, and END records. An overview of the PDB standard can be found here:</p>
<p><a class="reference external" href="http://www.wwpdb.org/documentation/file-format-content/format33/sect2.html#HEADER">http://www.wwpdb.org/documentation/file-format-content/format33/sect2.html#HEADER</a>
<a class="reference external" href="http://www.wwpdb.org/documentation/file-format-content/format33/sect8.html#CRYST1">http://www.wwpdb.org/documentation/file-format-content/format33/sect8.html#CRYST1</a>
<a class="reference external" href="http://www.wwpdb.org/documentation/file-format-content/format33/sect9.html#ATOM">http://www.wwpdb.org/documentation/file-format-content/format33/sect9.html#ATOM</a></p>
<p>PDB contains foure major parts; <code class="docutils literal notranslate"><span class="pre">REMARK</span></code>, <code class="docutils literal notranslate"><span class="pre">CRYST1</span></code>, <code class="docutils literal notranslate"><span class="pre">ATOM</span></code>, and <code class="docutils literal notranslate"><span class="pre">END</span></code>. Here is the definition of each field and how GOMC is using them to get the information it requires.</p>
<ul>
<li><p class="first"><code class="docutils literal notranslate"><span class="pre">REMARK</span></code>:
This header records present experimental  details, annotations, comments, and information not included in other records (for more information, <a class="reference external" href="http://www.wwpdb.org/documentation/file-format-content/format33/sect2.html#HEADER">click here</a>).</p>
<p>However, GOMC uses this header to print simulation informations.</p>
<ul class="simple">
<li><strong>Max Displacement</strong> (Å)</li>
<li><strong>Max Rotation</strong> (Degree)</li>
<li><strong>Max volume exchange</strong> (<span class="math notranslate nohighlight">\(Å^3\)</span>)</li>
<li><strong>Monte Carlo Steps</strong> (MC)</li>
</ul>
</li>
<li><p class="first"><code class="docutils literal notranslate"><span class="pre">CRYST1</span></code>:
This header records the unit <a class="reference external" href="http://www.wwpdb.org/documentation/file-format-content/format33/sect8.html#CRYST1">cell dimension parameters</a>.</p>
<ul class="simple">
<li><strong>Lattice constant</strong>: a,b,c (Å)</li>
<li><strong>Lattice angles</strong>: <span class="math notranslate nohighlight">\(\alpha, \beta, \gamma\)</span> (Degree)</li>
</ul>
</li>
<li><p class="first"><code class="docutils literal notranslate"><span class="pre">ATOM</span></code>:
The <a class="reference external" href="http://www.wwpdb.org/documentation/file-format-content/format33/sect9.html#ATOM">ATOM</a> records present the atomic coordinates for standard amino acids and nucleotides. They also present the occupancy and temperature factor for each atom.</p>
<ul class="simple">
<li><strong>ATOM</strong>: Record name</li>
<li><strong>serial</strong>: Atom serial number.</li>
<li><strong>name</strong>: Atom name.</li>
<li><strong>resName</strong>: Residue name.</li>
<li><strong>chainID</strong>: Chain identifier.</li>
<li><strong>resSeq</strong>: Residue sequence number.</li>
<li><strong>x</strong>: Coordinates for X (Å).</li>
<li><strong>y</strong>: Coordinates for Y (Å).</li>
<li><strong>z</strong>: Coordinates for Z (Å).</li>
<li><strong>occupancy</strong>: GOMC uses to define which atoms belong to which box.</li>
<li><strong>beta</strong>: Beta or Temperature factor. GOMC uses this value to define the mobility of the atoms. element: Element symbol.</li>
</ul>
</li>
<li><p class="first"><code class="docutils literal notranslate"><span class="pre">END</span></code>:
A frame in the PDB file is terminated with the keyword.</p>
</li>
</ul>
<p>Here are the PDB output of GOMC for the first molecule of isobutane:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>REMARK       GOMC    122.790    3.14159    3439.817     1000000
CRYST1     35.245     35.245     35.245       90.00       90.00     90.00
ATOM            1         C1        ISB           1       0.911    -0.313    0.000     0.00     0.00        C
ATOM            2         C1        ISB           1       1.424    -1.765    0.000     0.00     0.00        C
ATOM            3         C1        ISB           1      -0.629    -0.313    0.000     0.00     0.00        C
ATOM            4         C1        ISB           1       1.424     0.413   -1.257     0.00     0.00        C
END
</pre></div>
</div>
<p>The fields seen here in order from left to right are the record type, atom ID, atom name, residue name, residue ID, x, y, and z coordinates, occupancy, temperature factor (called beta), and segment name.</p>
<p>The atom name is “C1” and residue name is “ISB”. The PSF file (next section) contains a lookup table of atoms. These contain the atom name from the PDB and the name of the atom kind in the parameter file it corresponds to. As multiple different atom names will all correspond to the same parameter, these can be viewed “atom aliases” of sorts. The chain letter (in this case ‘A’) is sometimes used when packing a number of PDBs into a single PDB file.</p>
<div class="admonition important">
<p class="first admonition-title">Important</p>
<ul class="last simple">
<li>VMD requires a constant number of ATOMs in a multi-frame PDB (multiple records terminated by “END” in a single file). To compensate for this, all atoms from all boxes in the system are written to the output PDBs of this code.</li>
<li>For atoms not currently in a box, the coordinates are set to <code class="docutils literal notranslate"><span class="pre">&lt;</span> <span class="pre">0.00,</span> <span class="pre">0.00,</span> <span class="pre">0.00</span> <span class="pre">&gt;</span></code>. The occupancy is commonly just set to “1.00” and is left unused by many codes. We recycle this legacy parameter by using it to denote, in our output PDBs, the box a molecule is in (box 0 occupancy=0.00 ; box 1 occupancy=1.00)</li>
<li>The beta value in GOMC code is used to define the mobility of the molecule.<ul>
<li><code class="docutils literal notranslate"><span class="pre">Beta</span> <span class="pre">=</span> <span class="pre">0.00</span></code>: molecule can move and transfer within and between boxes.</li>
<li><code class="docutils literal notranslate"><span class="pre">Beta</span> <span class="pre">=</span> <span class="pre">1.00</span></code>: molecule is fixed in its position.</li>
<li><code class="docutils literal notranslate"><span class="pre">Beta</span> <span class="pre">=</span> <span class="pre">2.00</span></code>: molecule can move within the box but cannot be transferred between boxes.</li>
</ul>
</li>
</ul>
</div>
</div>
<div class="section" id="generating-pdb-file">
<h4>Generating PDB file<a class="headerlink" href="#generating-pdb-file" title="Permalink to this headline">¶</a></h4>
<p>With that overview of the format in mind, the following steps describe how a PDB file is typically built.</p>
<ol class="arabic simple">
<li>A single molecule PDB is obtained. In this example, the QM software package Gaussian was used to draw the molecule, which was then edited by hand to adhere to the PDB spec properly. The end result is a PDB for a single molecule:</li>
</ol>
<blockquote>
<div><div class="highlight-text notranslate"><div class="highlight"><pre><span></span>REMARK   1 File created by GaussView 5.0.8
ATOM     1  C1   ISB  1   0.911  -0.313    0.000  C
ATOM     2  C1   ISB  1   1.424  -1.765    0.000  C
ATOM     3  C1   ISB  1  -0.629  -0.313    0.000  C
ATOM     4  C1   ISB  1   1.424   0.413   -1.257  C
END
</pre></div>
</div>
</div></blockquote>
<ol class="arabic simple" start="2">
<li>Next, packings are calculated to place the simulation in a region of vapor-liquid coexistence. There are a couple of ways to do this in Gibbs ensemble:</li>
</ol>
<blockquote>
<div><ul class="simple">
<li>Pack both boxes to a single middle density, which is an average of the liquid and vapor densities.</li>
<li>Same as previous method, but add a modest amount to axis of one box (e.g. 10-30 A). This technique can be handy in the constant pressure Gibbs ensemble.</li>
<li>Pack one box to the predicted liquid density and the other to the vapor density.</li>
</ul>
<p>A good reference for getting the information needed to estimate packing is the NIST Web Book database of pure compounds:</p>
<p><a class="reference external" href="http://webbook.nist.gov/chemistry/">http://webbook.nist.gov/chemistry/</a></p>
</div></blockquote>
<ol class="arabic simple" start="3">
<li>After packing is determined, a basic pack can be performed with a Packmol script. Here is the example of packing 1000 isobutane in 70 A cubic box:</li>
</ol>
<blockquote>
<div><div class="highlight-text notranslate"><div class="highlight"><pre><span></span>tolerance   3.0
filetype    pdb
output      STEP2_ISB_packed_BOX 0.pdb
structure   isobutane.pdb
number      1000
inside cube 0.1   0.1   0.1   70.20
end     structure
</pre></div>
</div>
<p>Copy the above text into “pack_isobutane.inp” file, save it and run the script by typing the following line into the terminal:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ ./packmol &lt; pack_isobutane.inp
</pre></div>
</div>
</div></blockquote>
</div>
</div>
<div class="section" id="psf-file">
<h3>PSF File<a class="headerlink" href="#psf-file" title="Permalink to this headline">¶</a></h3>
<p>GOMC requires only one PSF file for NVT and NPT ensembles. However, GOMC requires two PSF files for GEMC and GCMC ensembles.</p>
<div class="section" id="what-is-psf-file">
<h4>What is PSF file<a class="headerlink" href="#what-is-psf-file" title="Permalink to this headline">¶</a></h4>
<p>Protein structure file (PSF), contains all of the molecule-specific information needed to apply a particular force field to a molecular system. The CHARMM force field is divided into a topology file, which is needed to generate the PSF file, and a parameter file, which supplies specific numerical values for the generic CHARMM potential function. The topology file defines the atom types used in the force field; the atom names, types, bonds, and partial charges of each residue type; and any patches necessary to link or otherwise mutate these basic residues. The parameter file provides a mapping between bonded and nonbonded interactions involving the various combinations of atom types found in the topology file and specific spring constants and similar parameters for all of the bond, angle, dihedral, improper, and van der Waals terms in the CHARMM potential function. PSF file widely used by by NAMD, CHARMM, and X-PLOR.</p>
<p>The PSF file contains six main sections: <code class="docutils literal notranslate"><span class="pre">remarks</span></code>, <code class="docutils literal notranslate"><span class="pre">atoms</span></code>, <code class="docutils literal notranslate"><span class="pre">bonds</span></code>, <code class="docutils literal notranslate"><span class="pre">angles</span></code>, <code class="docutils literal notranslate"><span class="pre">dihedrals</span></code>, and <code class="docutils literal notranslate"><span class="pre">impropers</span></code> (dihedral force terms used to maintain planarity). Each section starts with a specific header described bellow:</p>
<ul>
<li><p class="first"><code class="docutils literal notranslate"><span class="pre">NTITLE</span></code>: remarks on the file.
The following is taken from a PSF file for isobutane:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>PSF
      3  !NTITLE
REMARKS  original generated structure x-plor psf file
REMARKS  topology ./Top_Branched_Alkanes.inp
REMARKS  segment ISB { first NONE; last NONE; auto angles dihedrals }
</pre></div>
</div>
</li>
<li><p class="first"><code class="docutils literal notranslate"><span class="pre">NATOM</span></code>: Defines the atom names, types, and partial charges of each residue type.</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>atom    ID
segment name
residue ID
residue name
atom    name
atom    type
atom    charge
atom    mass
</pre></div>
</div>
<p>The following is taken from a PSF file for isobutane:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>4000 !NATOM
1    ISB  1  ISB    C1    CH1    0.000000   13.0190  0
2    ISB  1  ISB    C2    CH3    0.000000   15.0350  0
3    ISB  1  ISB    C3    CH3    0.000000   15.0350  0
4    ISB  1  ISB    C4    CH3    0.000000   15.0350  0
5    ISB  2  ISB    C1    CH1    0.000000   13.0190  0
6    ISB  2  ISB    C2    CH3    0.000000   15.0350  0
7    ISB  2  ISB    C3    CH3    0.000000   15.0350  0
8    ISB  2  ISB    C4    CH3    0.000000   15.0350  0
</pre></div>
</div>
<p>The fields in the atom section, from left to right are atom ID, segment name, residue ID, residue name, atom name, atom type, charge, mass, and an unused 0.</p>
</li>
<li><p class="first"><code class="docutils literal notranslate"><span class="pre">NBOND</span></code>: The covalent bond section lists four pairs of atoms per line. The following is taken from a PSF file for isobutane:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>3000   !BOND:     bonds
1   2   1   3   1   4   5   6
5   7   5   8
</pre></div>
</div>
</li>
<li><p class="first"><code class="docutils literal notranslate"><span class="pre">NTHETA</span></code>: The angle section lists three triples of atoms per line. The following is taken from a PSF file for isobutane:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>3000   !NTHETA:   angles
2   1   4   2   1   3   3   1   4
6   5   8   6   5   7   7   5   8
</pre></div>
</div>
</li>
<li><p class="first"><code class="docutils literal notranslate"><span class="pre">NPHI</span></code>: The dihedral sections list two quadruples of atoms per line.</p>
</li>
<li><p class="first"><code class="docutils literal notranslate"><span class="pre">NIMPHI</span></code>: The improper sections list two quadruples of atoms per line. GOMC currently does not support improper. For the molecules without dihedral or improper, PDF file look like the following:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>0   !NPHI: dihedrals
0   !NIMPHI: impropers
</pre></div>
</div>
</li>
<li><p class="first">(other sections such as cross terms)</p>
</li>
</ul>
<div class="admonition important">
<p class="first admonition-title">Important</p>
<ul class="last simple">
<li>The PSF file format is a highly redundant file format. It repeats identical topology of thousands of molecules of a common kind in some cases. GOMC follows the same approach as NAMD, allowing this excess information externally and compiling it in the code.</li>
<li>Other sections (e.g. cross terms) contain unsupported or legacy parameters and are ignored.</li>
<li>Following the restrictions of VMD, the order of the PSF atoms must match the order in the.</li>
<li>Improper entries are read and stored, but are not currently used. Support will eventually be added for this.</li>
</ul>
</div>
</div>
<div class="section" id="generating-psf-file">
<h4>Generating PSF file<a class="headerlink" href="#generating-psf-file" title="Permalink to this headline">¶</a></h4>
<p>The PSF file is typically generated using PSFGen. It is convenient to make a script, such as the example below, to do this:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>package require psfgen
topology  ./Top_branched_Alaknes.inp
segment ISB {
  pdb   ./STEP2_ISB_packed_BOX 0.pdb
  first   none
  last  none
}

coordpdb ./STEP2_ISB_packed_BOX 0.pdb ISB

writepsf ./STEP3_START_ISB_sys_BOX_0.psf
writepdb ./STEP3_START_ISB_sys_BOX_0.pdb
</pre></div>
</div>
<p>Typically, one script is run per box to generate a finalized PDB/PSF for that box. The script requires one additional file, the NAMD-style topology file. While GOMC does not directly read or interact with this file, it’s typically used to generate the PSF and, hence, is considered one of the integral file types. It will be briefly discussed in the following section.</p>
</div>
</div>
<div class="section" id="topology-file">
<h3>Topology File<a class="headerlink" href="#topology-file" title="Permalink to this headline">¶</a></h3>
<p>A CHARMM forcefield topology file contains all of the information needed to convert a list of residue names into a complete PSF structure file. The topology is a whitespace separated file format, which contains a list of atoms and their corresponding masses, and a list of residue information (charges, composition, and topology). Essentially, it is a non-redundant lookup table equivalent to the PSF file.</p>
<p>This is followed by a series of residues, which tell PSFGen what atoms are bonded to a given atom. Each residue is comprised of four key elements:</p>
<ul class="simple">
<li>A header beginning with the keyword RESI with the residue name and net charge</li>
<li>A body with multiple ATOM entries (not to be confused with the PDB-style entries of the same name), which list the partial charge on the particle and what kind of atom each named atom in a specific molecule/residue is.</li>
<li>A section of lines starting with the word BOND contains pairs of bonded atoms (typically 3 per line)</li>
<li>A closing section with instructions for PSFGen.</li>
</ul>
<p>Here’s an example of topology file for isobutane:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>* Custom top file -- branched alkanes *
11
!
MASS 1 CH3 15.035 C !
MASS 2 CH1 13.019 C !

AUTOGENERATE ANGLES DIHEDRALS

RESI ISB    0.00 !  isobutane - TraPPE
GROUP
ATOM  C1  CH1   0.00 !  C3\
ATOM  C2  CH3   0.00 !     C1-C2
ATOM  C3  CH3   0.00 !  C4/
ATOM  C4  CH3   0.00 !
BOND  C1  C2  C1  C3  C1  C4
PATCHING FIRS NONE LAST NONE

END
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The keyword END must be used to terminate this file and keywords related to the auto-generation process must be placed near the top of the file, after the MASS definitions.</p>
</div>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p>More in-depth information can be found in the following links:</p>
<ul class="simple">
<li><a class="reference external" href="http://www.ks.uiuc.edu/Training/Tutorials/science/topology/topology-tutorial.pdf">Topology Tutorial</a></li>
</ul>
<ul class="simple">
<li><a class="reference external" href="http://www.ks.uiuc.edu/Training/Tutorials/science/topology/topology-html/node4.html">NAMD Tutorial: Examining the Topology File</a></li>
</ul>
<ul class="simple">
<li><a class="reference external" href="http://www.ks.uiuc.edu/Training/Tutorials/science/forcefield-tutorial/forcefield-html/node6.html">Developing Topology and Parameter Files</a></li>
</ul>
<ul class="last simple">
<li><a class="reference external" href="http://www.ks.uiuc.edu/Training/Tutorials/namd/namd-tutorial-win-html/node25.html">NAMD Tutorial: Topology Files</a></li>
</ul>
</div>
</div>
<div class="section" id="parameter-file-s">
<h3>Parameter File(s)<a class="headerlink" href="#parameter-file-s" title="Permalink to this headline">¶</a></h3>
<p>Currently, GOMC uses a single parameter file and the user has the two kinds of parameter file choices:</p>
<ul class="simple">
<li><code class="docutils literal notranslate"><span class="pre">CHARMM</span></code> (Chemistry at Harvard Molecular Mechanics) compatible parameter file</li>
<li><code class="docutils literal notranslate"><span class="pre">EXOTIC</span></code> or <code class="docutils literal notranslate"><span class="pre">Mie</span></code> parameter file</li>
</ul>
<p>If the parameter file type is not specified or if the chosen file is missing, an error will result.</p>
<p>Both force field file options are whitespace separated files with sections preceded by a tag. When a known tag (representing a molecular interaction in the model) is encountered, reading of that section of the force field begins. Comments (anything after a <code class="docutils literal notranslate"><span class="pre">*</span></code> or <code class="docutils literal notranslate"><span class="pre">!</span></code>) and whitespace are ignored. Reading concludes when the end of the file is reached or another section tag is encountered.</p>
<div class="section" id="charmm-format-parameter-file">
<h4>CHARMM format parameter file<a class="headerlink" href="#charmm-format-parameter-file" title="Permalink to this headline">¶</a></h4>
<p>CHARMM contains a widely used model for describing energies in Monte Carlo and molecular dynamics simulations. It is intended to be compatible with other codes that use such a format, such as NAMD. See <a class="reference external" href="http://www.charmmtutorial.org/index.php/The_Energy_Function">here</a> for a general overview of the CHARMM force field.</p>
<p>Here’s the basic CHARMM contributions that are supported in GOMC:</p>
<div class="math notranslate nohighlight">
\[\begin{split}U_{\texttt{bond}}&amp;=\sum_{\texttt{bonds}} K_b(b-b_0)^2\\
U_{\texttt{dihedral}}&amp;=\sum_{\texttt{dihedrals}} K_{\phi} [1+\cos(n\phi - \delta)]\\
U_{\texttt{angle}}&amp;=\sum_{\texttt{angles}} K_{\theta}(\theta-\theta_0)^2\\
U_{\texttt{LJ}}&amp;=\sum_{\texttt{nonbonded}} \epsilon_{ij}\left[\left(\frac{R_{min_{ij}}}{r_{ij}}\right)^{12}-2\left(\frac{R_{min_{ij}}}{r_{ij}}\right)^6\right]+ \frac{q_i q_j}{\epsilon r_{ij}} \\\end{split}\]</div>
<p>As seen above, the following are recognized, read and used:</p>
<ul>
<li><p class="first"><code class="docutils literal notranslate"><span class="pre">BONDS</span></code>
- Quadratic expression describing bond stretching based on bond length (b) in Angstrom
– Typically, it is ignored as bonds are rigid for Monte Carlo simulations.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">GOMC does not sample bond stretch. To ignore the relative bond energy, set the <span class="math notranslate nohighlight">\(K_b\)</span> to a large value i.e. “999999999999”.</p>
</div>
<div class="figure" id="id1">
<img alt="_images/bonds.png" src="_images/bonds.png" />
<p class="caption"><span class="caption-text">Oscillations about the equilibrium bond length</span></p>
</div>
</li>
<li><p class="first"><code class="docutils literal notranslate"><span class="pre">ANGLES</span></code>
- Describe the conformational lbehavior of an angle (<span class="math notranslate nohighlight">\(\delta\)</span>) between three atoms, one of which is shared branch point to the other two.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">To fix any angle and ignore the related angle energy, set the <span class="math notranslate nohighlight">\(K_\theta\)</span> to a large value i.e. “999999999999”.</p>
</div>
<div class="figure" id="id2">
<img alt="_images/angle.png" src="_images/angle.png" />
<p class="caption"><span class="caption-text">Oscillations of 3 atoms about an equilibrium bond angle</span></p>
</div>
</li>
<li><p class="first"><code class="docutils literal notranslate"><span class="pre">DIHEDRALS</span></code>
- Describes crankshaft-like rotation behavior about a central bond in a series of three consecutive bonds (rotation is given as <span class="math notranslate nohighlight">\(\phi\)</span>).</p>
<div class="figure" id="id3">
<img alt="_images/dihedrals.png" src="_images/dihedrals.png" />
<p class="caption"><span class="caption-text">Torsional rotation of 4 atoms about a central bond</span></p>
</div>
</li>
<li><p class="first"><code class="docutils literal notranslate"><span class="pre">NONBONDED</span></code>
- This tag name only should be used if CHARMM force files are being used. This section describes 12-6 (Lennard-Jones) non-bonded interactions. Non-bonded parameters are assigned by specifying atom type name followed by polarizabilities (which will be ignored), minimum energy, and (minimum radius)/2. In order to modify 1-4 interaction, a second polarizability (again, will be ignored), minimum energy, and (minimum radius)/2 need to be defined; otherwise, the same parameter will be considered for 1-4 interaction.</p>
<div class="figure" id="id4">
<img alt="_images/nonbonded.png" src="_images/nonbonded.png" />
<p class="caption"><span class="caption-text">Non-bonded energy terms (electrostatics and Lennard-Jones)</span></p>
</div>
</li>
<li><p class="first"><code class="docutils literal notranslate"><span class="pre">NBFIX</span></code>
- This tag name only should be used if CHARMM force field is being used. This section allows in- teraction between two pairs of atoms to be modified, done by specifying two atom type names followed by minimum energy and minimum radius. In order to modify 1-4 interaction, a second minimum energy and minimum radius need to be defined; otherwise, the same parameter will be considered for 1-4 interaction.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Please pay attention that in this section we define minimum radius, not (minimum radius)/2 as it is defined in the NONBONDED section.</p>
</div>
<p>Currently, supported sections of the <code class="docutils literal notranslate"><span class="pre">CHARMM</span></code> compliant file include <code class="docutils literal notranslate"><span class="pre">BONDS</span></code>, <code class="docutils literal notranslate"><span class="pre">ANGLES</span></code>, <code class="docutils literal notranslate"><span class="pre">DIHEDRALS</span></code>, <code class="docutils literal notranslate"><span class="pre">NONBONDED</span></code>, <code class="docutils literal notranslate"><span class="pre">NBFIX</span></code>. Other sections such as <code class="docutils literal notranslate"><span class="pre">CMAP</span></code> are not currently read or supported.</p>
</li>
</ul>
</div>
<div class="section" id="bonds">
<h4>BONDS<a class="headerlink" href="#bonds" title="Permalink to this headline">¶</a></h4>
<p>(“bond stretching”) is one key section of the CHARMM-compliant file. Units for the <span class="math notranslate nohighlight">\(K_b\)</span> variable in this section are in kcal/mol; the <span class="math notranslate nohighlight">\(b_0\)</span> section (which represents the equilibrium bond length for that kind of pair) is measured in Angstroms.</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>BONDS
!V(bond) = Kb(b - b0)**2
!
!Kb:  kcal/mole/A**2
!b0:  A
!
!Kb (kcal/mol) = Kb (K) * Boltz.  const.;
!
!atom type      Kb          b0     description
CH3   CH1   9999999999    1.540 !  TraPPE 2
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The <span class="math notranslate nohighlight">\(K_b\)</span> value may appear odd, but this is because a larger value corresponds to a more rigid bond. As Monte Carlo force fields (e.g. TraPPE) typically treat molecules as rigid constructs, <span class="math notranslate nohighlight">\(K_b\)</span> is set to a large value - 9999999999. Sampling bond stretch is not supported in GOMC.</p>
</div>
</div>
<div class="section" id="angles">
<h4>ANGLES<a class="headerlink" href="#angles" title="Permalink to this headline">¶</a></h4>
<p>(“bond bending”), where <span class="math notranslate nohighlight">\(\theta\)</span> and <span class="math notranslate nohighlight">\(\theta_0\)</span> are commonly measured in degrees and <span class="math notranslate nohighlight">\(K_\theta\)</span> is measured in kcal/mol/K. These values, in literature, are often expressed in Kelvin (K). To convert Kelvin to kcal/mol/K, multiply by the Boltzmann constant – <span class="math notranslate nohighlight">\(K_\theta\)</span>, 0.0019872041 kcal/mol. In order to fix the angle, it requires to set a large value for <span class="math notranslate nohighlight">\(K_\theta\)</span>. By assigning a large value like 9999999999, specified angle will be fixed and energy of that angle will considered to be zero.</p>
<p>Here is an example of what is necessary for isobutane:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>ANGLES
!
!V(angle) = Ktheta(Theta - Theta0)**2
!
!V(Urey-Bradley) = Kub(S - S0)**2
!
!Ktheta:  kcal/mole/rad**2
!Theta0:  degrees
!S0:  A
!
!Ktheta (kcal/mol) = Ktheta (K) * Boltz.  const.
!
!atom types         Ktheta        Theta0
CH3   CH1   CH3     62.100125     112.00 !  TraPPE 2
</pre></div>
</div>
<p>Some CHARMM ANGLES section entries include <code class="docutils literal notranslate"><span class="pre">Urey-Bradley</span></code> potentials (<span class="math notranslate nohighlight">\(K_{ub}\)</span>, <span class="math notranslate nohighlight">\(b_{ub}\)</span>), in addition to the standard quadratic angle potential. The constants related to this potential function are currently read, but the logic has not been added to calculate this potential function. Support for this potential function will be added in later versions of the code.</p>
</div>
<div class="section" id="dihedrals">
<h4>DIHEDRALS<a class="headerlink" href="#dihedrals" title="Permalink to this headline">¶</a></h4>
<p>The final major bonded interactions section of the CHARMM compliant parameter file are the DIHEDRALS. Each dihedral is composed of a dihedral series of 1 or more terms. Often, there are 4 to 6 terms in a dihedral. Angles for the dihedrals’ deltas are given in degrees.</p>
<p>Since isobutane has no dihedral, here are the parameters pertaining to 2,3-dimethylbutane:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>DIHEDRALS
!
!V(dihedral) = Kchi(1 + cos(n(chi) - delta))
!
!Kchi:  kcal/mole
!n:  multiplicity
!delta:  degrees
!
!Kchi (kcal/mol) = Kchi (K) * Boltz.  const.
!
!atom types             Kchi    n     delta   description
X   CH1   CH1   X    -0.498907  0     0.0   !  TraPPE 2
X   CH1   CH1   X     0.851974  1     0.0   !  TraPPE 2
X   CH1   CH1   X    -0.222269  2   180.0   !  TraPPE 2
X   CH1   CH1   X     0.876894  3     0.0   !  TraPPE 2
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The code allows the use of ‘X’ to indicate ambiguous positions on the ends. This is useful because this kind is often determined solely by the two middle atoms in the middle of the dihedral, according to literature.</p>
</div>
</div>
<div class="section" id="impropers">
<h4>IMPROPERS<a class="headerlink" href="#impropers" title="Permalink to this headline">¶</a></h4>
<p>Energy parameters used to describe out-of-plane rocking are currently read, but unused. The section is often blank. If it becomes necessary, algorithms to calculate the improper energy will need to be added.</p>
</div>
<div class="section" id="nonbonded">
<h4>NONBONDED<a class="headerlink" href="#nonbonded" title="Permalink to this headline">¶</a></h4>
<p>The next section of the CHARMM style parameter file is the NONBONDED. In order to use TraPPE this section of the CHARMM compliant file is critical. Here’s an example with our isobutane potential model:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>NONBONDED
!
!V(Lennard-Jones) = Eps,i,j[(Rmin,i,j/ri,j)**12 - 2(Rmin,i,j/ri,j)**6]
!
!atom ignored epsilon         Rmin/2        ignored   eps,1-4     Rmin/2,1-4
CH3   0.0     -0.194745992  2.10461634058     0.0       0.0       0.0 !  TraPPE 1
CH1   0.0     -0.019872040  2.62656119304     0.0       0.0       0.0 !  TraPPE 2
End
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The <span class="math notranslate nohighlight">\(R_{min}\)</span> is the potential well-depth, where the attraction is maximum. However, <span class="math notranslate nohighlight">\(\sigma\)</span> is the particle diameter, where the interaction energy is zero. To convert <span class="math notranslate nohighlight">\(\sigma\)</span> to <span class="math notranslate nohighlight">\(R_{min}\)</span>, simply multiply <span class="math notranslate nohighlight">\(\sigma\)</span> by 0.56123102415.</p>
</div>
<div class="admonition important">
<p class="first admonition-title">Important</p>
<p class="last">If no parameter was defined for 1-4 interaction e.g (<span class="math notranslate nohighlight">\(\epsilon_{1-4}, Rmin_{1-4}/2\)</span>), GOMC will use the  <span class="math notranslate nohighlight">\(\epsilon, Rmin/2\)</span> for 1-4 interaction.</p>
</div>
</div>
<div class="section" id="nbfix">
<h4>NBFIX<a class="headerlink" href="#nbfix" title="Permalink to this headline">¶</a></h4>
<p>The last section of the CHARMM style parameter file is the NBFIX. In this section, individual pair interaction will be modified. First, pseudo non-bonded parameters have to be defined in NONBONDED and modified in NBFIX. Here iss an example if it is required to modify interaction between CH3 and CH1 atoms:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>NBFIX
!V(Lennard-Jones) = Eps,i,j[(Rmin,i,j/ri,j)**12 - 2(Rmin,i,j/ri,j)**6]
!
!atom atom  epsilon         Rmin          eps,1-4   Rmin,1-4
CH3   CH1   -0.294745992    1.10461634058 !
End
</pre></div>
</div>
<div class="admonition important">
<p class="first admonition-title">Important</p>
<p class="last">If no parameter was defined for 1-4 interaction e.g (<span class="math notranslate nohighlight">\(\epsilon_{1-4}, Rmin_{1-4}\)</span>), GOMC will use the  <span class="math notranslate nohighlight">\(\epsilon, Rmin\)</span> for 1-4 interaction.</p>
</div>
</div>
</div>
<div class="section" id="exotic-or-mie-parameter-file">
<h3>Exotic or Mie Parameter File<a class="headerlink" href="#exotic-or-mie-parameter-file" title="Permalink to this headline">¶</a></h3>
<p>The Mie file is intended for use with nonstandard/specialty models of molecular interaction, which are not included in CHARMM standard. Currently, two custom interaction are included:</p>
<ul class="simple">
<li><code class="docutils literal notranslate"><span class="pre">NONBODED_MIE</span></code> This section describes n-6 (Lennard-Jones) non-bonded interactions. The Lennard- Jones potential (12-6) is a subset of this potential. Non-bonded parameters are assigned by specifying atom type name followed by minimum energy, atom diameter, and repulsion exponent. In order to modify 1-4 interaction, a second minimum energy, atom diameter, and repulsion exponent need to be defined; otherwise, the same parameters would be considered for 1-4 interaction.</li>
<li><code class="docutils literal notranslate"><span class="pre">NBFIX_MIE</span></code> This section allows n-6 (Lennard-Jones) interaction between two pairs of atoms to be modified. This is done by specifying two atoms type names followed by minimum energy, atom diameter, and repulsion exponent. In order to modify 1-4 interaction, a second minimum energy, atom diameter, and repulsion exponent need to be defined.</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">In <code class="docutils literal notranslate"><span class="pre">Mie</span></code> force field, the definition of atom diameter(<span class="math notranslate nohighlight">\(\sigma\)</span>) is same for both <code class="docutils literal notranslate"><span class="pre">NONBONDED_MIE</span></code> and <code class="docutils literal notranslate"><span class="pre">NBFIX_MIE</span></code>.</p>
</div>
<div class="admonition important">
<p class="first admonition-title">Important</p>
<p class="last">If no parameter was defined for 1-4 interaction e.g (<span class="math notranslate nohighlight">\(\epsilon_{1-4}, \sigma_{1-4}, n_{1-4}\)</span>), GOMC will use the  <span class="math notranslate nohighlight">\(\epsilon, \sigma, n\)</span> for 1-4 interaction.</p>
</div>
<p>Otherwise, the Mie file reuses the same geometry section headings - BONDS / ANGLES / DIHEDRALS / etc. The only difference in these sections versus in the CHARMM format force field file is that the energies are in Kelvin (‘K’), the unit most commonly found for parameters in Monte Carlo chemical simulation literature. This precludes the need to convert to kcal/mol, the energy unit used in CHARMM.
The most frequently used section of the Mie files in the Mie potential section is NONBONDED_MIE. Here are the parameters that are used to simulate alkanes:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>NONBONDED_MIE
!
!V(mie) = const*eps*((sig ij/r ij)^n-(sig ij/r ij)^6)
!
!atom eps       sig     n     eps,1-4   sig,1-4   n,1-4
CH4   161.00    3.740   14    0.0       0.0       0.0 ! Potoff, et al. &#39;09
CH3   121.25    3.783   16    0.0       0.0       0.0 ! Potoff, et al. &#39;09
CH2    61.00    3.990   16    0.0       0.0       0.0 ! Potoff, et al. &#39;09
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Although the units (Angstroms) are the same, the Mie file uses <span class="math notranslate nohighlight">\(\sigma\)</span>, not the <span class="math notranslate nohighlight">\(R_{min}\)</span> used by CHARMM. The energy in the exotic file are expressed in Kelvin (K), as this is the standard convention in the literature.</p>
</div>
</div>
<div class="section" id="control-file-conf">
<h3>Control File (*.conf)<a class="headerlink" href="#control-file-conf" title="Permalink to this headline">¶</a></h3>
<p>The control file is GOMC’s proprietary input file. It contains key settings. The settings generally fall under three categories:</p>
<ul class="simple">
<li>Input/Simulation Setup</li>
<li>System Settings for During Run</li>
<li>Output Settings</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The control file is designed to recognize logic values, such as “yes/true/on” or “no/false/off”. The keyword in control file is not case sensitive.</p>
</div>
<div class="section" id="input-simulation-setup">
<h4>Input/Simulation Setup<a class="headerlink" href="#input-simulation-setup" title="Permalink to this headline">¶</a></h4>
<p>In this section, input file names are listed. In addition, if you want to restart your simulation or use integer seed for running your simulation, you need to modify this section according to your purpose.</p>
<dl class="docutils">
<dt><code class="docutils literal notranslate"><span class="pre">Restart</span></code></dt>
<dd><p class="first">Determines whether to restart the simulation from previous simulation or not.</p>
<ul class="last simple">
<li>Value 1: Boolean - True if restart, false otherwise.</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">PRNG</span></code></dt>
<dd><p class="first">Dictates how to start the pseudo-random number generator (PRNG)</p>
<ul>
<li><p class="first">Value 1: String</p>
<ul class="simple">
<li>RANDOM: Randomizes Mersenne Twister PRNG with random bits based on the system time.</li>
</ul>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# kind {RANDOM, INTSEED}
#################################
PRNG   RANDOM
</pre></div>
</div>
<ul class="simple">
<li>INTSEED: This option “seeds” the Mersenne Twister PRNG with a standard integer. When the same integer is used, the generated PRNG stream should be the same every time, which is helpful in tracking down bugs.</li>
</ul>
</li>
</ul>
<dl class="last docutils">
<dt><code class="docutils literal notranslate"><span class="pre">Random_Seed</span></code></dt>
<dd><p class="first">Defines the seed number. If “INTSEED” is chosen, seed number needs to be specified; otherwise, the program will terminate.</p>
<ul class="simple">
<li>Value 1: ULONG - If “INTSEED” option is selected for PRNG (See above example)</li>
</ul>
<div class="last highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# kind {RANDOM, INTSEED}
#################################
PRNG          INTSEED
RandomSeed    50
</pre></div>
</div>
</dd>
</dl>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">ParaTypeCHARMM</span></code></dt>
<dd><p class="first">Sets force field type to CHARMM style.</p>
<ul class="simple">
<li>Value 1: Boolean - True if it is CHARMM forcefield, false otherwise.</li>
</ul>
<div class="last highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# FORCE FIELD TYPE
#################################
ParaTypeCHARMM    true
</pre></div>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">ParaTypeEXOTIC</span></code> or <code class="docutils literal notranslate"><span class="pre">ParaTypeMie</span></code></dt>
<dd><p class="first">Sets force field type to Mie style.</p>
<ul class="simple">
<li>Value 1: Boolean - True if it is Mie forcefield, false otherwise.</li>
</ul>
<div class="last highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# FORCE FIELD TYPE
#################################
ParaTypeEXOTIC    true
</pre></div>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">ParaTypeMARTINI</span></code></dt>
<dd><p class="first">Sets force field type to MARTINI style.</p>
<ul class="simple">
<li>Value 1: Boolean - True if it is MARTINI forcefield, false otherwise.</li>
</ul>
<div class="last highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# FORCE FIELD TYPE
#################################
ParaTypeMARTINI     true
</pre></div>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">Parameters</span></code></dt>
<dd><p class="first">Provides the name and location of the parameter file to use for the simulation.</p>
<ul class="simple">
<li>Value 1: String - Sets the name of the parameter file.</li>
</ul>
<div class="last highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# FORCE FIELD TYPE
#################################
ParaTypeCHARMM    yes
Parameters        ../../common/Par_TraPPE_Alkanes.inp
</pre></div>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">Coordinates</span></code></dt>
<dd><p class="first">Defines the PDB file names (coordinates) and location for each box in the system.</p>
<ul class="simple">
<li>Value 1: Integer - Sets box number (starts from ‘0’).</li>
<li>Value 2: String - Sets the name of PDB file.</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">NVT and NPT ensembles requires only one PDB file and GEMC/GCMC requires two PDB files. If the number of PDB files is not compatible with the simulation type, the program will terminate.</p>
</div>
<p>Example of NVT or NPT ensemble:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>#############################################
# INPUT PDB FILES - NVT or NPT ensemble
#############################################
Coordinates   0   STEP3_START_ISB_sys.pdb
</pre></div>
</div>
<p>Example of Gibbs or GC ensemble:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>#############################################
# INPUT PDB FILES - Gibbs or GCMC ensemble
#############################################
Coordinates   0   STEP3_START_ISB_sys_BOX_0.pdb
Coordinates   1   STEP3_START_ISB_sys_BOX_1.pdb
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">In case of <code class="docutils literal notranslate"><span class="pre">Restart</span></code> true, the restart PDB output file from GOMC (<code class="docutils literal notranslate"><span class="pre">OutputName</span></code> _BOX_0_restart.pdb) can be used for each box.</p>
</div>
<p>Example of Gibbs ensemble when Restart mode is active:</p>
<div class="last highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# INPUT PDB FILES
#################################
Coordinates   0   ISB_T_270_k_BOX_0_restart.pdb
Coordinates   1   ISB_T_270_k_BOX_1_restart.pdb
</pre></div>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">Structures</span></code></dt>
<dd><p class="first">Defines the PSF filenames (structures) for each box in the system.</p>
<ul class="simple">
<li>Value 1: Integer - Sets box number (start from ‘0’)</li>
<li>Value 2: String - Sets the name of PSF file.</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">NVT and NPT ensembles requires only one PSF file and GEMC/GCMC requires two PSF files. If the number of PSF files is not compatible with the simulation type, the program will terminate.</p>
</div>
<p>Example of NVT or NPT ensemble:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# INPUT PSF FILES
#################################
Structure   0   STEP3_START_ISB_sys.psf
</pre></div>
</div>
<p>Example of Gibbs or GC ensemble:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# INPUT PSF FILES
#################################
Structure   0   STEP3_START_ISB_sys_BOX_0.psf
Structure   1   STEP3_START_ISB_sys_BOX_1.psf
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">In case of <code class="docutils literal notranslate"><span class="pre">Restart</span></code> true, the PSF output file from GOMC (<code class="docutils literal notranslate"><span class="pre">OutputName</span></code> _merged.psf) can be used for both boxes.</p>
</div>
<p>Example of Gibbs ensemble when <code class="docutils literal notranslate"><span class="pre">Restart</span></code> mode is active:</p>
<div class="last highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# INPUT PSF FILES
#################################
Structure   0   ISB_T_270_k_merged.psf
Structure   1   ISB_T_270_k_merged.psf
</pre></div>
</div>
</dd>
</dl>
</div>
<div class="section" id="system-settings-for-during-run-setup">
<h4>System Settings for During Run Setup<a class="headerlink" href="#system-settings-for-during-run-setup" title="Permalink to this headline">¶</a></h4>
<p>This section contains all the variables not involved in the output of data during the simulation, or in the reading of input files at the start of the simulation. In other words, it contains settings related to the moves, the thermodynamic constants (based on choice of ensemble), and the length of the simulation.
Note that some tags, or entries for tags, are only used in certain ensembles (e.g. Gibbs ensemble). These cases are denoted with colored text.</p>
<dl class="docutils">
<dt><code class="docutils literal notranslate"><span class="pre">GEMC</span></code></dt>
<dd><p class="first"><em>(For Gibbs Ensemble runs only)</em> Defines the type of Gibbs Ensemble simulation you want to run. If neglected in Gibbs Ensemble, it simply defaults to const volume (NVT) Gibbs Ensemble.</p>
<ul class="simple">
<li>Value 1: String - Allows you to pick between isovolumetric (“NVT”) and isobaric (“NPT”) Gibbs ensemble simulations.</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The default value for <code class="docutils literal notranslate"><span class="pre">GEMC</span></code> is NVT.</p>
</div>
<div class="last highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# GEMC TYPE (DEFAULT IS NVT GEMC)
#################################
GEMC    NVT
</pre></div>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">Pressure</span></code></dt>
<dd><p class="first">For <code class="docutils literal notranslate"><span class="pre">NPT</span></code> or <code class="docutils literal notranslate"><span class="pre">NPT-GEMC</span></code> simulation, imposed pressure (in bar) needs to be specified; otherwise, the program will terminate.</p>
<ul class="simple">
<li>Value 1: Double - Constant pressure in bar.</li>
</ul>
<div class="last highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# GEMC TYPE (DEFAULT IS NVT GEMC)
#################################
GEMC        NPT
Pressure    5.76
</pre></div>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">Temperature</span></code></dt>
<dd><p class="first">Sets the temperature at which the system will run.</p>
<ul class="last simple">
<li>Value 1: Double - Constant temperature of simulation in degrees Kelvin.</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">Rcut</span></code></dt>
<dd><p class="first">Sets a specific radius that non-bonded interaction energy and force will be considered and calculated using defined potential function.</p>
<ul class="last simple">
<li>Value 1: Double - The distance to truncate the Lennard-Jones potential at.</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">RcutLow</span></code></dt>
<dd><p class="first">Sets a specific minimum possible in angstrom that reject any move that places any atom closer than specified distance.</p>
<ul class="last simple">
<li>Value 1: Double - The minimum possible distance between any atoms.</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">RcutCoulomb</span></code></dt>
<dd><p class="first">Sets a specific radius for each box in the system that short range electrostatic energy will be calculated.</p>
<ul class="simple">
<li>Value 1: Integer - Sets box number (start from ‘0’)</li>
<li>Value 2: Double - The distance to truncate the short rage electrostatic energy at.</li>
</ul>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The default value for <code class="docutils literal notranslate"><span class="pre">RcutCoulomb</span></code> is the value of <code class="docutils literal notranslate"><span class="pre">Rcut</span></code></p>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">LRC</span></code></dt>
<dd><p class="first">Defines whether or not long range corrections are used.</p>
<ul class="simple">
<li>Value 1: Boolean - True to consider long range correction.</li>
</ul>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<p class="last">In case of using <code class="docutils literal notranslate"><span class="pre">SHIFT</span></code> or <code class="docutils literal notranslate"><span class="pre">SWITCH</span></code> potential functions, <code class="docutils literal notranslate"><span class="pre">LRC</span></code> will be ignored.</p>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">Exclude</span></code></dt>
<dd><p class="first">Defines which pairs of bonded atoms should be excluded from non-bonded interactions.</p>
<ul class="last">
<li><p class="first">Value 1: String - Allows you to choose between “1-2”, “1-3”, and “1-4”.</p>
<ul class="simple">
<li>1-2: All interactions pairs of bonded atoms, except the ones that separated with one bond, will be considered and modified using 1-4 parameters defined in parameter file.</li>
<li>1-3: All interaction pairs of bonded atoms, except the ones that separated with one or two bonds, will be considered and modified using 1-4 parameters defined in parameter file.</li>
<li>1-4: All interaction pairs of bonded atoms, except the ones that separated with one, two or three bonds, will be considered using non-bonded parameters defined in parameter file.</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The default value for <code class="docutils literal notranslate"><span class="pre">Exclude</span></code> is “1-4”.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">In CHARMM force field, the 1-4 interaction needs to be considered. Choosing “<code class="docutils literal notranslate"><span class="pre">Exclude</span></code> 1-3” will modify 1-4 interaction based on 1-4 parameters in parameter file. If a kind force field is used, where 1-4 interaction needs to be ignored, such as TraPPE, either “<code class="docutils literal notranslate"><span class="pre">Exclude</span></code> 1-4” needs to be chosen or 1-4 parameter needs to be assigned to zero in the parameter file.</p>
</div>
</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">Potential</span></code></dt>
<dd><p class="first">Defines the potential function type to calculate non-bonded interaction energy and force between atoms.</p>
<ul class="last">
<li><p class="first">Value 1: String - Allows you to pick between “VDW”, “SHIFT” and “SWITCH”.</p>
<ul>
<li><p class="first">VDW: Nonbonded interaction energy and force calculated based on n-6 (Lennard-Johns) equation. This function will be discussed further in the Intermolecular energy and Virial calculation section.</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# SIMULATION CONDITION
#################################
Temperature   270.00
Potential     VDW
LRC           true
Rcut          10
Exclude       1-4
</pre></div>
</div>
</li>
<li><p class="first">SHIFT: This option forces the potential energy to be zero at <code class="docutils literal notranslate"><span class="pre">Rcut</span></code> distance. This function will be discussed further in the Intermolecular energy and Virial calculation section.</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# SIMULATION CONDITION
#################################
Temperature     270.00
Potential       SHIFT
LRC             false
Rcut            10
Exclude         1-4
RcutCoulomb  0  12.0
RcutCoulomb  1  20.0
</pre></div>
</div>
</li>
<li><p class="first">SWITCH: This option smoothly forces the potential energy to be zero at <code class="docutils literal notranslate"><span class="pre">Rcut</span></code> distance and starts modifying the potential at <code class="docutils literal notranslate"><span class="pre">Rswitch</span></code> distance. Depending on force field type, specific potential function will be applied. These functions will be discussed further in the Intermolecular energy and Virial calculation section.</p>
</li>
</ul>
<dl class="docutils">
<dt><code class="docutils literal notranslate"><span class="pre">Rswitch</span></code></dt>
<dd><p class="first">In the case of choosing “SWITCH” as potential function, a distance is set in which non-bonded interaction energy is truncated smoothly at <code class="docutils literal notranslate"><span class="pre">Rcut</span></code> distance.</p>
<ul class="last simple">
<li>Value 1: Double - Define switch distance in angstrom. If the “SWITCH” function is chosen, <code class="docutils literal notranslate"><span class="pre">Rswitch</span></code> needs to be defined; otherwise, the program will be terminated.</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">VDWGeometricSigma</span></code></dt>
<dd><p class="first">Use geometric mean, as required by OPLS force field, to combining Lennard-Jones sigma parameters for different atom types.</p>
<ul class="last">
<li><p class="first">Value 1: Boolean - True, uses geometric mean to combine L-J sigmas</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The default setting of <code class="docutils literal notranslate"><span class="pre">VDWGeometricSigma</span></code> is false to use arithmetic mean when combining Lennard-Jones sigma parameters for different atom types.</p>
</div>
</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">ElectroStatic</span></code></dt>
<dd><p class="first">Considers coulomb interaction or not. This function will be discussed further in the Inter- molecular energy and Virial calculation section.</p>
<ul class="last">
<li><p class="first">Value 1: Boolean - True if coulomb interaction needs to be considered and false if not.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">To simulate the polar molecule in MARTINI force field, <code class="docutils literal notranslate"><span class="pre">ElectroStatic</span></code> needs to be turn on. MARTINI force field uses short range coulomb interaction with constant <code class="docutils literal notranslate"><span class="pre">Dielectric</span></code> 15.0.</p>
</div>
</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">Ewald</span></code></dt>
<dd><p class="first">Considers standard Ewald summation method for electrostatic calculation. This function will be discussed further in the Intermolecular energy and Virial calculation section.</p>
<ul class="last">
<li><p class="first">Value 1: Double - True if Ewald summation calculation needs to be considered and false if not.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">By default, <code class="docutils literal notranslate"><span class="pre">ElectroStatic</span></code> will be set to true if Ewald summation method was used to calculate coulomb interaction.</p>
</div>
</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">CachedFourier</span></code></dt>
<dd><p class="first">Considers storing the reciprocal terms for Ewald summation calculation in order to improve the code performance. This option would increase the code performance with the cost of memory usage.</p>
<ul class="last">
<li><p class="first">Value 1: Boolean - True to store reciprocal terms of Ewald summation calculation and false if not.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">By default, <code class="docutils literal notranslate"><span class="pre">CachedFourier</span></code> will be set to true if not value was set.</p>
</div>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">Monte Carlo moves, such as <code class="docutils literal notranslate"><span class="pre">MEMC-1</span></code>, <code class="docutils literal notranslate"><span class="pre">MEMC-2</span></code>, <code class="docutils literal notranslate"><span class="pre">MEMC-3</span></code>, <code class="docutils literal notranslate"><span class="pre">IntraMEMC-1</span></code>, <code class="docutils literal notranslate"><span class="pre">IntraMEMC-2</span></code>, <code class="docutils literal notranslate"><span class="pre">IntraMEMC-3</span></code> does not support <code class="docutils literal notranslate"><span class="pre">CachedFourier</span></code>.</p>
</div>
</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">Tolerance</span></code></dt>
<dd><p class="first">Specifies the accuracy of the Ewald summation calculation. Ewald separation parameter and number of reciprocal vectors for the Ewald summation are determined based on the accuracy parameter.</p>
<ul class="last">
<li><p class="first">Value 1: Double - Sets the accuracy in Ewald summation calculation.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>A reasonable value for te accuracy is 0.00001.</li>
<li>If “Ewald” was chosen and no value was set for Tolerance, the program will be terminated.</li>
</ul>
</div>
</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">Dielectric</span></code></dt>
<dd><p class="first">Defines dielectric constant for coulomb interaction in MARTINI force field.</p>
<ul class="last">
<li><p class="first">Value 1: Double - Sets dielectric value used in coulomb interaction.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>In MARTINI force field, <code class="docutils literal notranslate"><span class="pre">Dielectric</span></code> needs to be set to 15.0.</li>
<li>If MARTINI force field was chosen and <code class="docutils literal notranslate"><span class="pre">Dielectric</span></code> was not specified, a default value of 15.0 will be assigned.</li>
</ul>
</div>
</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">PressureCalc</span></code></dt>
<dd><p class="first">Considers to calculate the pressure or not. If it is set to true, the frequency of pressure calculation need to be set.</p>
<ul class="last simple">
<li>Value 1: Boolean - True enabling pressure calculation during the simulation, false disabling pressure calculation.</li>
<li>Value 2: Ulong - The frequency of calculating the pressure.</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">1-4scaling</span></code></dt>
<dd><p class="first">Defines constant factor to modify intra-molecule coulomb interaction.</p>
<ul class="last">
<li><p class="first">Value 1: Double - A fraction number between 0.0 and 1.0.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">CHARMM force field uses a value between 0.0 and 1.0. In MARTINI force field, it needs to be set to 1.0 because 1-4 interaction will not be modified in this force field.</p>
</div>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# SIMULATION CONDITION
#################################
ElectroStatic   true
Ewald           true
Tolerance       0.00001
CachedFourier   false
1-4scaling      0.0
</pre></div>
</div>
</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">RunSteps</span></code></dt>
<dd><p class="first">Sets the total number of steps to run (one move is performed for each step) (cycles = this value / number of molecules in the system)</p>
<ul class="simple">
<li>Value 1: Ulong - Total run steps</li>
</ul>
<div class="last admonition important">
<p class="first admonition-title">Important</p>
<p class="last">Seting the <code class="docutils literal notranslate"><span class="pre">RunSteps</span></code> to zero, and activating <code class="docutils literal notranslate"><span class="pre">Restart</span></code> simulation, will recalculate the energy of stored simulation’s snapshots.</p>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">EqSteps</span></code></dt>
<dd><p class="first">Sets the number of steps necessary to equilibrate the system; averaging will begin at this step.</p>
<ul class="simple">
<li>Value 1: Ulong - Equilibration steps</li>
</ul>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<p class="last">In GCMC simulation, the <code class="docutils literal notranslate"><span class="pre">Histogram</span></code> files will be dumped at <code class="docutils literal notranslate"><span class="pre">EqSteps</span></code>.</p>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">AdjSteps</span></code></dt>
<dd><p class="first">Sets the number of steps per adjustment of the parameter associated with each move (e.g. maximum translate distance, maximum rotation, maximum volume exchange, etc.)</p>
<ul class="last">
<li><p class="first">Value 1: Ulong - Number of steps per move adjustment</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# STEPS
#################################
RunSteps    25000000
EqSteps     5000000
AdjSteps    1000
</pre></div>
</div>
</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">ChemPot</span></code></dt>
<dd><p class="first">For Grand Canonical (GC) ensemble runs only: Chemical potential at which simulation is run.</p>
<ul class="simple">
<li>Value 1: String - The residue name to apply this chemical potential.</li>
<li>Value 2: Double - The chemical potential value in degrees Kelvin (should be negative).</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>For binary systems, include multiple copies of the tag (one per residue kind).</li>
<li>If there is a molecule kind that cannot be transfer between boxes (in PDB file the beta value is set to 1.00 or 2.00), an arbitrary value (e.g. 0.00) can be assigned to the residue name.</li>
</ul>
</div>
<div class="last highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# Mol.  Name Chem.  Pot.  (K)
#################################
ChemPot   ISB     -968
</pre></div>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">Fugacity</span></code></dt>
<dd><p class="first">For Grand Canonical (GC) ensemble runs only: Fugacity at which simulation is run.</p>
<ul class="simple">
<li>Value 1: String - The residue to apply this fugacity.</li>
<li>Value 2: Double - The fugacity value in bar.</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>For binary systems, include multiple copies of the tag (one per residue kind).</li>
<li>If there is a molecule kind that cannot be transfer between boxes (in PDB file the beta value is set to 1.00 or 2.00) an arbitrary value e.g. 0.00 can be assigned to the residue name.</li>
</ul>
</div>
<div class="last highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# Mol.  Name Fugacity (bar)
#################################
Fugacity  ISB   10.0
Fugacity  Si     0.0
Fugacity  O      0.0
</pre></div>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">DisFreq</span></code></dt>
<dd><p class="first">Fractional percentage at which displacement move will occur.</p>
<ul class="last simple">
<li>Value 1: Double - % Displacement</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">RotFreq</span></code></dt>
<dd><p class="first">Fractional percentage at which rigid rotation move will occur.</p>
<ul class="last simple">
<li>Value 1: Double - % Rotatation</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">IntraSwapFreq</span></code></dt>
<dd><p class="first">Fractional percentage at which molecule will be removed from a box and inserted into the same box using coupled-decoupled configurational-bias algorithm.</p>
<ul class="simple">
<li>Value 1: Double - % Intra molecule swap</li>
</ul>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The default value for <code class="docutils literal notranslate"><span class="pre">IntraSwapFreq</span></code> is 0.000</p>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">RegrowthFreq</span></code></dt>
<dd><p class="first">Fractional percentage at which part of the molecule will be deleted and then regrown using coupled-decoupled configurational-bias algorithm.</p>
<ul class="simple">
<li>Value 1: Double - % Molecular growth</li>
</ul>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The default value for <code class="docutils literal notranslate"><span class="pre">RegrowthFreq</span></code> is 0.000</p>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">CrankShaftFreq</span></code></dt>
<dd><p class="first">Fractional percentage at which crankshaft move will occur. In this move, two atoms that are forming angle or dihedral are selected randomely and form a shaft. Then any atoms or group that are within these two selected atoms, will rotate around the shaft to sample intramolecular degree of freedom.</p>
<ul class="simple">
<li>Value 1: Double - % Crankshaft</li>
</ul>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The default value for <code class="docutils literal notranslate"><span class="pre">CrankShaftFreq</span></code> is 0.000</p>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">IntraMEMC-1Freq</span></code></dt>
<dd><p class="first">Fractional percentage at which specified number of small molecule kind will be exchanged with a specified large molecule kind in defined sub-volume within same simulation box.</p>
<ul class="simple">
<li>Value 1: Double - % Molecular exchange</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>The default value for <code class="docutils literal notranslate"><span class="pre">IntraMEMC-1Freq</span></code> is 0.000</li>
<li>This move need additional information such as <code class="docutils literal notranslate"><span class="pre">ExchangeVolumeDim</span></code>, <code class="docutils literal notranslate"><span class="pre">ExchangeRatio</span></code>, <code class="docutils literal notranslate"><span class="pre">ExchangeSmallKind</span></code>, and <code class="docutils literal notranslate"><span class="pre">ExchangeLargeKind</span></code>, which will be explained later.</li>
<li>For more information about this move, please refere to <a class="reference external" href="https://aip.scitation.org/doi/abs/10.1063/1.5025184">MEMC-GCMC</a> and <a class="reference external" href="https://www.sciencedirect.com/science/article/pii/S0378381218305351">MEMC-GEMC</a> papers.</li>
</ul>
</div>
<dl class="docutils">
<dt><code class="docutils literal notranslate"><span class="pre">IntraMEMC-2Freq</span></code></dt>
<dd>Fractional percentage at which specified number of small molecule kind will be exchanged with a specified large molecule kind in defined sub-volume within same simulation box. Backbone of small and large molecule kind will be used to insert the large molecule more efficiently.</dd>
</dl>
<ul class="simple">
<li>Value 1: Double - % Molecular exchange</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>The default value for <code class="docutils literal notranslate"><span class="pre">IntraMEMC-2Freq</span></code> is 0.000</li>
<li>This move need additional information such as <code class="docutils literal notranslate"><span class="pre">ExchangeVolumeDim</span></code>, <code class="docutils literal notranslate"><span class="pre">ExchangeRatio</span></code>, <code class="docutils literal notranslate"><span class="pre">ExchangeSmallKind</span></code>, <code class="docutils literal notranslate"><span class="pre">ExchangeLargeKind</span></code>, <code class="docutils literal notranslate"><span class="pre">SmallKindBackBone</span></code>, and <code class="docutils literal notranslate"><span class="pre">LargeKindBackBone</span></code>, which will be explained later.</li>
<li>For more information about this move, please refere to <a class="reference external" href="https://aip.scitation.org/doi/abs/10.1063/1.5025184">MEMC-GCMC</a> and <a class="reference external" href="https://www.sciencedirect.com/science/article/pii/S0378381218305351">MEMC-GEMC</a> papers.</li>
</ul>
</div>
<dl class="docutils">
<dt><code class="docutils literal notranslate"><span class="pre">IntraMEMC-3Freq</span></code></dt>
<dd>Fractional percentage at which specified number of small molecule kind will be exchanged with a specified large molecule kind in defined sub-volume within same simulation box. Specified atom of the large molecule kind will be used to insert the large molecule using coupled-decoupled configurational-bias.</dd>
</dl>
<ul class="simple">
<li>Value 1: Double - % Molecular exchange</li>
</ul>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>The default value for <code class="docutils literal notranslate"><span class="pre">IntraMEMC-3Freq</span></code> is 0.000</li>
<li>This move need additional information such as <code class="docutils literal notranslate"><span class="pre">ExchangeVolumeDim</span></code>, <code class="docutils literal notranslate"><span class="pre">ExchangeRatio</span></code>, <code class="docutils literal notranslate"><span class="pre">ExchangeSmallKind</span></code>, <code class="docutils literal notranslate"><span class="pre">ExchangeLargeKind</span></code>, and <code class="docutils literal notranslate"><span class="pre">LargeKindBackBone</span></code>, which will be explained later.</li>
<li>For more information about this move, please refere to <a class="reference external" href="https://aip.scitation.org/doi/abs/10.1063/1.5025184">MEMC-GCMC</a> and <a class="reference external" href="https://www.sciencedirect.com/science/article/pii/S0378381218305351">MEMC-GEMC</a> papers.</li>
</ul>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">MEMC-1Freq</span></code></dt>
<dd><p class="first">For Gibbs and Grand Canonical (GC) ensemble runs only: Fractional percentage at which specified number of small molecule kind will be exchanged with a specified large molecule kind in defined sub-volume in dense simulation box.</p>
<ul class="simple">
<li>Value 1: Double - % Molecular exchange</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>The default value for <code class="docutils literal notranslate"><span class="pre">IntraMEMC-1Freq</span></code> is 0.000</li>
<li>This move need additional information such as <code class="docutils literal notranslate"><span class="pre">ExchangeVolumeDim</span></code>, <code class="docutils literal notranslate"><span class="pre">ExchangeRatio</span></code>, <code class="docutils literal notranslate"><span class="pre">ExchangeSmallKind</span></code>, and <code class="docutils literal notranslate"><span class="pre">ExchangeLargeKind</span></code>, which will be explained later.</li>
<li>For more information about this move, please refere to <a class="reference external" href="https://aip.scitation.org/doi/abs/10.1063/1.5025184">MEMC-GCMC</a> and <a class="reference external" href="https://www.sciencedirect.com/science/article/pii/S0378381218305351">MEMC-GEMC</a> papers.</li>
</ul>
</div>
<dl class="docutils">
<dt><code class="docutils literal notranslate"><span class="pre">MEMC-2Freq</span></code></dt>
<dd>For Gibbs and Grand Canonical (GC) ensemble runs only: Fractional percentage at which specified number of small molecule kind will be exchanged with a specified large molecule kind in defined sub-volume in dense simulation box. Backbone of small and large molecule kind will be used to insert the large molecule more efficiently.</dd>
</dl>
<ul class="simple">
<li>Value 1: Double - % Molecular exchange</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>The default value for <code class="docutils literal notranslate"><span class="pre">IntraMEMC-2Freq</span></code> is 0.000</li>
<li>This move need additional information such as <code class="docutils literal notranslate"><span class="pre">ExchangeVolumeDim</span></code>, <code class="docutils literal notranslate"><span class="pre">ExchangeRatio</span></code>, <code class="docutils literal notranslate"><span class="pre">ExchangeSmallKind</span></code>, <code class="docutils literal notranslate"><span class="pre">ExchangeLargeKind</span></code>, <code class="docutils literal notranslate"><span class="pre">SmallKindBackBone</span></code>, and <code class="docutils literal notranslate"><span class="pre">LargeKindBackBone</span></code>, which will be explained later.</li>
<li>For more information about this move, please refere to <a class="reference external" href="https://aip.scitation.org/doi/abs/10.1063/1.5025184">MEMC-GCMC</a> and <a class="reference external" href="https://www.sciencedirect.com/science/article/pii/S0378381218305351">MEMC-GEMC</a> papers.</li>
</ul>
</div>
<dl class="docutils">
<dt><code class="docutils literal notranslate"><span class="pre">MEMC-3Freq</span></code></dt>
<dd>For Gibbs and Grand Canonical (GC) ensemble runs only: Fractional percentage at which specified number of small molecule kind will be exchanged with a specified large molecule kind in defined sub-volume in dense simulation box. Specified atom of the large molecule kind will be used to insert the large molecule using coupled-decoupled configurational-bias.</dd>
</dl>
<ul class="simple">
<li>Value 1: Double - % Molecular exchange</li>
</ul>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>The default value for <code class="docutils literal notranslate"><span class="pre">IntraMEMC-3Freq</span></code> is 0.000</li>
<li>This move need additional information such as <code class="docutils literal notranslate"><span class="pre">ExchangeVolumeDim</span></code>, <code class="docutils literal notranslate"><span class="pre">ExchangeRatio</span></code>, <code class="docutils literal notranslate"><span class="pre">ExchangeSmallKind</span></code>, <code class="docutils literal notranslate"><span class="pre">ExchangeLargeKind</span></code>, and <code class="docutils literal notranslate"><span class="pre">LargeKindBackBone</span></code>, which will be explained later.</li>
<li>For more information about this move, please refere to <a class="reference external" href="https://aip.scitation.org/doi/abs/10.1063/1.5025184">MEMC-GCMC</a> and <a class="reference external" href="https://www.sciencedirect.com/science/article/pii/S0378381218305351">MEMC-GEMC</a> papers.</li>
</ul>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">SwapFreq</span></code></dt>
<dd><p class="first">For Gibbs and Grand Canonical (GC) ensemble runs only: Fractional percentage at which molecule swap move will occur using coupled-decoupled configurational-bias.</p>
<ul class="last simple">
<li>Value 1: Double - % Molecule swaps</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">VolFreq</span></code></dt>
<dd><p class="first">For isobaric-isothermal ensemble and Gibbs ensemble runs only: Fractional percentage at which molecule will be removed from one box and inserted into the other box using configurational bias algorithm.</p>
<ul class="last simple">
<li>Value 1: Double - % Volume swaps</li>
</ul>
</dd>
</dl>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# MOVE FREQEUNCY
#################################
DisFreq         0.39
RotFreq         0.10
IntraSwapFreq   0.10
RegrowthFreq    0.10
CrankShaftFreq  0.10
SwapFreq        0.20
VolFreq         0.01
</pre></div>
</div>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">All move percentages should add up to 1.0; otherwise, the program will terminate.</p>
</div>
<dl class="docutils">
<dt><code class="docutils literal notranslate"><span class="pre">ExchangeVolumeDim</span></code></dt>
<dd><p class="first">To use all variation of <code class="docutils literal notranslate"><span class="pre">MEMC</span></code> and <code class="docutils literal notranslate"><span class="pre">IntraMEMC</span></code> Monte Carlo moves, the exchange sub-volume must be defined. The exchange sub-volume is defined as an orthogonal box with x-, y-, and z-dimensions, where small molecule/molecules kind will be selected from to be exchanged with a large molecule kind.</p>
<ul class="simple">
<li>Value 1: Double - X dimension in <span class="math notranslate nohighlight">\(Å\)</span></li>
<li>Value 2: Double - Y dimension in <span class="math notranslate nohighlight">\(Å\)</span></li>
<li>Value 3: Double - Z dimension in <span class="math notranslate nohighlight">\(Å\)</span></li>
</ul>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>Currently, the X and Y dimension cannot be set independently (X = Y = max(X, Y))</li>
<li>A heuristic for setting good values of the x-, y-, and z-dimensions is to use the geometric size of the large molecule plus 1-2 Å in each dimension.</li>
<li>In case of exchanging 1 small molecule kind with 1 large molecule kind in <code class="docutils literal notranslate"><span class="pre">IntraMEMC-2</span></code>, <code class="docutils literal notranslate"><span class="pre">IntraMEMC-3</span></code>, <code class="docutils literal notranslate"><span class="pre">MEMC-2</span></code>, <code class="docutils literal notranslate"><span class="pre">MEMC-3</span></code> Monte Carlo moves, the sub-volume dimension has no effect on acceptance rate.</li>
</ul>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">ExchangeSmallKind</span></code></dt>
<dd><p class="first">To use all variation of <code class="docutils literal notranslate"><span class="pre">MEMC</span></code> and <code class="docutils literal notranslate"><span class="pre">IntraMEMC</span></code> Monte Carlo moves, the small molecule kind to be exchanged with a large molecule kind must be defined. Multiple small molecule kind can be specified.</p>
<ul class="last simple">
<li>Value 1: String - Small molecule kind to be exchanged.</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">ExchangeLargeKind</span></code></dt>
<dd><p class="first">To use all variation of <code class="docutils literal notranslate"><span class="pre">MEMC</span></code> and <code class="docutils literal notranslate"><span class="pre">IntraMEMC</span></code> Monte Carlo moves, the large molecule kind to be exchanged with small molecule kind must be defined. Multiple large molecule kind can be specified.</p>
<ul class="last simple">
<li>Value 1: String - Large molecule kind to be exchanged.</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">ExchangeRatio</span></code></dt>
<dd><p class="first">To use all variation of <code class="docutils literal notranslate"><span class="pre">MEMC</span></code> and <code class="docutils literal notranslate"><span class="pre">IntraMEMC</span></code> Monte Carlo moves, the exchange ratio must be defined. The exchange ratio defines how many small molecule will be exchanged with 1 large molecule. For each large-small molecule pairs, one exchange ratio must be defined.</p>
<ul class="last simple">
<li>Value 1: Integer - Ratio of exchanging small molecule/molecules with 1 large molecule.</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">LargeKindBackBone</span></code></dt>
<dd><p class="first">To use <code class="docutils literal notranslate"><span class="pre">MEMC-2</span></code>, <code class="docutils literal notranslate"><span class="pre">MEMC-3</span></code>, <code class="docutils literal notranslate"><span class="pre">IntraMEMC-2</span></code>, and <code class="docutils literal notranslate"><span class="pre">IntraMEMC-3</span></code> Monte Carlo moves, the large molecule backbone must be defined. The backbone of the molecule is defined as a vector that connects two atoms belong to the large molecule. The large molecule backbone will be used to align the sub-volume in <code class="docutils literal notranslate"><span class="pre">MEMC-2</span></code> and <code class="docutils literal notranslate"><span class="pre">IntraMEMC-2</span></code> moves, while in <code class="docutils literal notranslate"><span class="pre">MEMC-3</span></code> and <code class="docutils literal notranslate"><span class="pre">IntraMEMC-3</span></code> moves, it uses the atom name to start growing the large molecule using coupled-decoupled configurational-bias. For each large-small molecule pairs, two atom names must be defined.</p>
<ul class="simple">
<li>Value 1: String - Atom name 1 belong to the large molecule’s backbone</li>
<li>Value 2: String - Atom name 2 belong to the large molecule’s backbone</li>
</ul>
<div class="last admonition important">
<p class="first admonition-title">Important</p>
<p class="last">In <code class="docutils literal notranslate"><span class="pre">MEMC-3</span></code> and <code class="docutils literal notranslate"><span class="pre">IntraMEMC-3</span></code> Monte Carlo moves, both atom names must be same, otherwise program will be terminated.</p>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">SmallKindBackBone</span></code></dt>
<dd><p class="first">To use <code class="docutils literal notranslate"><span class="pre">MEMC-2</span></code>, and <code class="docutils literal notranslate"><span class="pre">IntraMEMC-2</span></code> Monte Carlo moves, the small molecule backbone must be defined. The backbone of the molecule is defined as a vector that connects two atoms belong to the small molecule and will be used to align the sub-volume. For each large-small molecule pairs, two atom names must be defined.</p>
<ul class="last simple">
<li>Value 1: String - Atom name 1 belong to the small molecule’s backbone</li>
<li>Value 2: String - Atom name 2 belong to the small molecule’s backbone</li>
</ul>
</dd>
</dl>
<p>Here is the example of <code class="docutils literal notranslate"><span class="pre">MEMC-2</span></code> Monte Carlo moves, where 7 large-small molecule pairs are defined with an exchange ratio of 1:1: (ethane, methane), (propane, ethane), (n-butane, propane), (n-pentane, nbutane), (n-hexane, n-pentane), (n-heptane, n-hexane), and (noctane, n-heptane).</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>######################################################################
# MEMC PARAMETER
######################################################################
ExchangeVolumeDim   1.0   1.0   1.0
ExchangeRatio       1       1       1      1      1      1      1
ExchangeLargeKind   C8P    C7P    C6P    C5P    C4P    C3P    C2P
ExchangeSmallKind   C7P    C6P    C5P    C4P    C3P    C2P    C1P
LargeKindBackBone   C1 C8  C1 C7  C1 C6  C1 C5  C1 C4  C1 C3  C1 C2
SmallKindBackBone   C1 C7  C1 C6  C1 C5  C1 C4  C1 C3  C1 C2  C1 C1
</pre></div>
</div>
<dl class="docutils">
<dt><code class="docutils literal notranslate"><span class="pre">useConstantArea</span></code></dt>
<dd><p class="first">For Isobaric-Isothermal ensemble and Gibbs ensemble runs only: Considers to change the volume of the simulation box by fixing the cross-sectional area (x-y plane).</p>
<ul class="simple">
<li>Value 1: Boolean - If true volume will change only in z axis, If false volume will change with constant axis ratio.</li>
</ul>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<p class="last">By default, <code class="docutils literal notranslate"><span class="pre">useConstantArea</span></code> will be set to false if no value was set. It means, the volume of the box will change in a way to maintain the constant axis ratio.</p>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">FixVolBox0</span></code></dt>
<dd><p class="first">For adsorption simulation in NPT Gibbs ensemble runs only: Changing the volume of fluid phase (Box 1) to maintain the constant imposed pressure and temperature, while keeping the volume of adsorbed phase (Box 0) fix.</p>
<ul class="last simple">
<li>Value 1: Boolean - If true volume of adsorbed phase will remain constant, If false volume of adsorbed phase will change.</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">CellBasisVector</span></code></dt>
<dd><p class="first">Defines the shape and size of the simulation periodic cell. <code class="docutils literal notranslate"><span class="pre">CellBasisVector1</span></code>, <code class="docutils literal notranslate"><span class="pre">CellBasisVector2</span></code>, <code class="docutils literal notranslate"><span class="pre">CellBasisVector3</span></code> represent the cell basis vector <span class="math notranslate nohighlight">\(a,b,c\)</span>, respectively. This tag may occur multiple times. It occurs once for NVT and NPT, but twice for Gibbs ensemble or GC ensemble.</p>
<ul class="simple">
<li>Value 1: Integer - Sets box number (first box is box ‘0’).</li>
<li>Value 2: Double - x value of cell basis vector <span class="math notranslate nohighlight">\(Å\)</span>.</li>
<li>Value 3: Double - y value of cell basis vector <span class="math notranslate nohighlight">\(Å\)</span>.</li>
<li>Value 4: Double - z value of cell basis vector <span class="math notranslate nohighlight">\(Å\)</span>.</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">If the number of defined boxes were not compatible to simulation type, the program will be terminated.</p>
</div>
<p>Example for NVT and NPT ensemble. In this example, each vector is perpendicular to the other two (<span class="math notranslate nohighlight">\(\alpha = 90, \beta = 90, \gamma = 90\)</span>), as indicated by a single x, y, or z value being specified by each and making a rectangular 3-D box:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>############################################
# BOX DIMENSION #, X, Y, Z
############################################
CellBasisVector1  0   40.00   00.00   00.00
CellBasisVector2  0   00.00   40.00   00.00
CellBasisVector3  0   00.00   00.00   80.00
</pre></div>
</div>
<p>Example for Gibbs ensemble and GC ensemble ensemble. In this example, In the first box, only vector <span class="math notranslate nohighlight">\(a\)</span> and <span class="math notranslate nohighlight">\(c\)</span> are perpendicular to each other (<span class="math notranslate nohighlight">\(\alpha = 90, \beta = 90, \gamma = 120\)</span>), and making a non-orthogonal simulation cell with the cell length <span class="math notranslate nohighlight">\(a = 39.91 Å, b = 39.91 Å, c = 76.98 Å\)</span>. In the second box, each vector is perpendicular to the other two (<span class="math notranslate nohighlight">\(\alpha = 90, \beta = 90, \gamma = 90\)</span>), as indicated by a single x, y, or z value being specified by each and making a cubic box:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>############################################
# BOX DIMENSION #, X, Y, Z
############################################
CellBasisVector1  0   36.91   00.00   00.00
CellBasisVector2  0   -18.45  31.96   00.00
CellBasisVector3  0   00.00   00.00   76.98

CellBasisVector1  1   60.00   00.00   00.00
CellBasisVector2  1   00.00   60.00   00.00
CellBasisVector3  1   00.00   00.00   60.00
</pre></div>
</div>
<div class="last admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">If <code class="docutils literal notranslate"><span class="pre">Restart</span></code> was activated, box dimension does not need to be specified. If it is specified, program will read it but it will be ignored and replaced by the printed cell dimensions and angles in the restart PDB output file from GOMC (<code class="docutils literal notranslate"><span class="pre">OutputName_BOX_0_restart.pdb</span></code> and <code class="docutils literal notranslate"><span class="pre">Output_Name_BOX_1_restart.pdb</span></code>).</p>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">CBMC_First</span></code></dt>
<dd><p class="first">Number of CD-CBMC trials to choose the first atom position (Lennard-Jones trials for first seed growth).</p>
<ul class="last simple">
<li>Value 1: Integer - Number of initial insertion sites to try.</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">CBMC_Nth</span></code></dt>
<dd><p class="first">Number of CD-CBMC trials to choose the later atom positions (Lennard-Jones trials for first seed growth).</p>
<ul class="last simple">
<li>Value 1: Integer - Number of LJ trials for growing later atom positions.</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">CBMC_Ang</span></code></dt>
<dd><p class="first">Number of CD-CBMC bending angle trials to perform for geometry (per the coupled-decoupled CBMC scheme).</p>
<ul class="last simple">
<li>Value 1: Integer - Number of trials per angle.</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">CBMC_Dih</span></code></dt>
<dd><p class="first">Number of CD-CBMC dihedral angle trials to perform for geometry (per the coupled-decoupled CBMC scheme).</p>
<ul class="simple">
<li>Value 1: Integer - Number of trials per dihedral.</li>
</ul>
<div class="last highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# CBMC TRIALS
#################################
CBMC_First  10
CBMC_Nth    4
CBMC_Ang    100
CBMC_Dih    30
</pre></div>
</div>
</dd>
</dl>
</div>
<div class="section" id="output-controls">
<h4>Output Controls<a class="headerlink" href="#output-controls" title="Permalink to this headline">¶</a></h4>
<p>This section contains all the values that control output in the control file. For example, certain variables control the naming of files dumped of the block-averaged thermodynamic variables of interest, the PDB files, etc.</p>
<dl class="docutils">
<dt><code class="docutils literal notranslate"><span class="pre">OutputName</span></code></dt>
<dd><p class="first">Unique name with no space for simulation used to name the block average, PDB, and PSF output files.</p>
<ul class="simple">
<li>Value 1: String - Unique phrase to identify this system.</li>
</ul>
<div class="last highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# OUTPUT FILE NAME
#################################
OutputName  ISB_T_270_K
</pre></div>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">CoordinatesFreq</span></code></dt>
<dd><p class="first">Controls output of PDB file (coordinates). If PDB dumping was enabled, one file for NVT or NPT and two files for Gibbs ensemble or GC ensemble will be dumped into <code class="docutils literal notranslate"><span class="pre">OutputName_BOX_n.pdb</span></code>, where n defines the box number.</p>
<ul class="simple">
<li>Value 1: Boolean - “true” enables dumping these files; “false” disables dumping.</li>
<li>Value 2: Ulong - Steps per dump PDB frame. It should be less than or equal to RunSteps. If this keyword could not be found in configuration file, its value will be assigned a default value to dump 10 frames.</li>
</ul>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>The PDB file contains an entry for every ATOM, in all boxes read. This allows VMD (which requires a constant number of atoms) to properly parse frames, with a bit of help. Atoms that are not currently in a specific box are given the coordinate (0.00, 0.00, 0.00). The occupancy value corresponds to the box a molecule is currently in (e.g. 0.00 for box 0; 1.00 for box 1).</li>
<li>At the beginning of simulation, a merged PSF file will be dumped into <code class="docutils literal notranslate"><span class="pre">OutputName</span></code> _merged.pdb, in which all boxes will be dumped. It also contains the topology for every molecule in both boxes, corresponding to the merged PDB format. Loading PDB files into merged PSF file in VMD allows the user to visualize and analyze the results.</li>
<li>In addition, this file can be used to load into GOMC once <code class="docutils literal notranslate"><span class="pre">Restart</span></code> simulation was active and <code class="docutils literal notranslate"><span class="pre">RunSteps</span></code> sets to 0 to recalculate the energy of stored snapshot of the previous simulation.</li>
</ul>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">RestartFreq</span></code></dt>
<dd><p class="first">Controls the output of the last state of simulation at a specified step in PDB files (coordinates) OutputName BOX n restart.pdb, where n defines the box number. Header part of this file contains important information and will be needed to restart the simulation:</p>
<ul class="simple">
<li>Simulation cell dimensions and angles.</li>
<li>Maximum amount of displacement (Å), rotation (<span class="math notranslate nohighlight">\(\delta\)</span>), and volume (<span class="math notranslate nohighlight">\(Å^3\)</span>) that used in Displacement, Rotation, and Volume move.</li>
</ul>
<p>If PDB dumping was enabled, one file for NVT or NPT and two files for Gibbs ensemble or GC ensemble will be dumped.</p>
<ul class="simple">
<li>Value 1: Boolean - “true” enables dumping these files; “false” disables dumping.</li>
<li>Value 2: Ulong - Steps per dump last state of simulation to PDB files. It should be less than or equal to RunSteps. If this keyword could not be found in the configuration file, RestartFreq value will be assigned by default.</li>
</ul>
<div class="last admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>The restart PDB file contains only ATOM that exist in each boxes at specified steps. This allows the user to load this file into GOMC once <code class="docutils literal notranslate"><span class="pre">Restart</span></code> simulation was active.</li>
<li>CoordinatesFreq must be a common multiple of RestartFreq or vice versa.</li>
</ul>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">ConsoleFreq</span></code></dt>
<dd><p class="first">Controls the output to STDIO (“the console”) of messages such as acceptance statistics, and run timing info. In addition, instantaneously-selected thermodynamic properties will be output to this file.</p>
<ul class="last simple">
<li>Value 1: Boolean - “true” enables message printing; “false” disables dumping.</li>
<li>Value 2: Ulong - Number of steps per print. If this keyword could not be found in the configuration file, the value will be assigned by default to dump 1000 output for RunSteps greater than 1000 steps and 100 output for RunSteps less than 1000 steps.</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">BlockAverageFreq</span></code></dt>
<dd><p class="first">Controls the block averages output of selected thermodynamic properties. Block averages are averages of thermodynamic values of interest for chunks of the simulation (for post-processing of averages or std. dev. in those values).</p>
<ul class="last simple">
<li>Value 1: Boolean - “true” enables printing block average; “false” disables it.</li>
<li>Value 2: Ulong - Number of steps per block-average output file. If this keyword cannot be found in the configuration file, its value will be assigned a default to dump 100 output.</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">HistogramFreq</span></code></dt>
<dd><p class="first">Controls the histograms. Histograms are a binned listing of observation frequency for a specific thermodynamic variable. In this code, they also control the output of a file containing energy/molecule samples; it only will be used in GC ensemble simulations for histogram reweighting purposes.</p>
<ul class="simple">
<li>Value 1: Boolean - “true” enables printing histogram; “false” disables it.</li>
<li>Value 2: Ulong - Number of steps per histogram output file. If this keyword cannot be found in the configuration file, a value will be assigned by default to dump 1000 output for RunSteps greater than 1000 steps and 100 output for RunSteps less than 1000 steps.</li>
</ul>
<div class="last highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# STATISTICS Enable, Freq.
#################################
CoordinatesFreq   true 10000000
RestartFreq       true 1000000
ConsoleFreq       true 100000
BlockAverageFreq  true 100000
HistogramFreq     true 10000
</pre></div>
</div>
</dd>
</dl>
<p>The next section controls the output of the energy/molecule sample file and the distribution file for molecule counts, commonly referred to as the “histogram” output. This section is only required if Grand Canonical ensemble simulation was used.</p>
<dl class="docutils">
<dt><code class="docutils literal notranslate"><span class="pre">DistName</span></code></dt>
<dd><p class="first">Sets short phrase to naming molecule distribution file.</p>
<ul class="last simple">
<li>Value 1: String - Short phrase which will be combined with <em>RunNumber</em> and <em>RunLetter</em> to use in the name of the binned histogram for molecule distribution.</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">HistName</span></code></dt>
<dd><p class="first">Sets short phrase to naming energy sample file.</p>
<ul class="last simple">
<li>Value 1: String - Short phrase, which will be combined with <em>RunNumber</em> and <em>RunLetter</em>, to use in the name of the energy/molecule count sample file.</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">RunNumber</span></code></dt>
<dd><p class="first">Sets a number, which is a part of <em>DistName</em> and <em>HistName</em> file name.</p>
<ul class="last simple">
<li>Value 1: Uint – Run number to be used in the above file names.</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">RunLetter</span></code></dt>
<dd><p class="first">Sets a letter, which is a part of <em>DistName</em> and <em>HistName</em> file name.</p>
<ul class="last simple">
<li>Value 1: Character – Run letter to be used in above file names.</li>
</ul>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">SampleFreq</span></code></dt>
<dd><p class="first">Controls histogram sampling frequency.</p>
<ul class="simple">
<li>Value 1: Uint – the number of steps per histogram sample.</li>
</ul>
<div class="last highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# OutHistSettings
#################################
DistName   dis
HistName   his
RunNumber  5
RunLetter  a
SampleFreq 200
</pre></div>
</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">OutEnergy,</span> <span class="pre">OutPressure,</span> <span class="pre">OutMolNumber,</span> <span class="pre">OutDensity,</span> <span class="pre">OutVolume,</span> <span class="pre">OutSurfaceTension</span></code></dt>
<dd><p class="first">Enables/Disables for specific kinds of file output for tracked thermodynamic quantities</p>
<ul class="simple">
<li>Value 1: Boolean – “true” enables message output of block averages via this tracked parameter (and in some cases such as entry, components); “false” disables it.</li>
<li>Value 2: Boolean – “true” enables message output of a fluctuation into the console file via this tracked parameter (and in some cases, such as entry, components); “false” disables it.</li>
</ul>
<p>The keywords are available for the following ensembles</p>
<table border="1" class="docutils">
<colgroup>
<col width="25%" />
<col width="25%" />
<col width="25%" />
<col width="25%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Keyword</th>
<th class="head">NVT</th>
<th class="head">NPT &amp; Gibbs</th>
<th class="head">GC</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>OutEnergy</td>
<td><span class="math notranslate nohighlight">\(\checkmark\)</span></td>
<td><span class="math notranslate nohighlight">\(\checkmark\)</span></td>
<td><span class="math notranslate nohighlight">\(\checkmark\)</span></td>
</tr>
<tr class="row-odd"><td>OutPressure</td>
<td><span class="math notranslate nohighlight">\(\checkmark\)</span></td>
<td><span class="math notranslate nohighlight">\(\checkmark\)</span></td>
<td><span class="math notranslate nohighlight">\(\checkmark\)</span></td>
</tr>
<tr class="row-even"><td>OutMolNumber</td>
<td>&#160;</td>
<td><span class="math notranslate nohighlight">\(\checkmark\)</span></td>
<td><span class="math notranslate nohighlight">\(\checkmark\)</span></td>
</tr>
<tr class="row-odd"><td>OutDensity</td>
<td>&#160;</td>
<td><span class="math notranslate nohighlight">\(\checkmark\)</span></td>
<td><span class="math notranslate nohighlight">\(\checkmark\)</span></td>
</tr>
<tr class="row-even"><td>OutVolume</td>
<td>&#160;</td>
<td><span class="math notranslate nohighlight">\(\checkmark\)</span></td>
<td><span class="math notranslate nohighlight">\(\checkmark\)</span></td>
</tr>
<tr class="row-odd"><td>OutSurfaceTension</td>
<td><span class="math notranslate nohighlight">\(\checkmark\)</span></td>
<td>&#160;</td>
<td>&#160;</td>
</tr>
</tbody>
</table>
<p>Here is an example:</p>
<div class="last highlight-text notranslate"><div class="highlight"><pre><span></span>#################################
# ENABLE: BLK AVE., FLUC.
#################################
OutEnergy         true true
OutPressure       true true
OutMolNum         true true
OutDensity        true true
OutVolume         true true
OutSurfaceTention false false
</pre></div>
</div>
</dd>
</dl>
</div>
</div>
</div>
<span id="document-output_file"></span><div class="section" id="gomc-s-output-files-terminal-output">
<h2>GOMC’s Output Files, Terminal Output<a class="headerlink" href="#gomc-s-output-files-terminal-output" title="Permalink to this headline">¶</a></h2>
<dl class="docutils">
<dt>GOMC currently supports several kinds of output:</dt>
<dd><ul class="first last">
<li><p class="first">STDIO (“console”) output</p>
</li>
<li><p class="first">File output</p>
<p>– PDB</p>
<p>– PSF</p>
<p>– Block Averages</p>
</li>
</ul>
</dd>
</dl>
<p>GOMC output units:</p>
<table border="1" class="docutils">
<colgroup>
<col width="57%" />
<col width="43%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Properties</th>
<th class="head">Units</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Energy</td>
<td><span class="math notranslate nohighlight">\(K\)</span></td>
</tr>
<tr class="row-odd"><td>Pressure, Pressure Tensor</td>
<td>bar</td>
</tr>
<tr class="row-even"><td>Heat of vaporization</td>
<td><span class="math notranslate nohighlight">\(KJ/mol\)</span></td>
</tr>
<tr class="row-odd"><td>Volume</td>
<td><span class="math notranslate nohighlight">\(Å^3\)</span></td>
</tr>
<tr class="row-even"><td>Density</td>
<td><span class="math notranslate nohighlight">\(kg/m^3\)</span></td>
</tr>
<tr class="row-odd"><td>Mol Density</td>
<td><span class="math notranslate nohighlight">\(molecule/Å^3\)</span></td>
</tr>
<tr class="row-even"><td>Surface Tension</td>
<td><span class="math notranslate nohighlight">\(mN/m\)</span></td>
</tr>
</tbody>
</table>
<div class="section" id="console-output">
<h3>Console Output<a class="headerlink" href="#console-output" title="Permalink to this headline">¶</a></h3>
<p>A variety of useful information relating to instantaneous statistical and thermodynamic data (move trials, acceptance rates, file I/O messages warnings, and other kinds of information) is printed to the STDIO, which, in Linux, will typically be displayed in the terminal. This output can be redirected into a log file in Linux using the <code class="docutils literal notranslate"><span class="pre">&gt;</span></code> operator.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ GOMC CPU NVT in.conf &gt; out_isobutane.log <span class="p">&amp;</span>
</pre></div>
</div>
<p>Statistical and thermodynamic information is provided in console output.</p>
<ul>
<li><p class="first">Energy</p>
<p>– Intermolecular (LJ)</p>
<p>– Intramolecular bonded</p>
<p>– Intramolecular nonbonded</p>
<p>– Tail corrections</p>
<p>– Electrostatic real</p>
<p>– Electrostatic Reciprocal</p>
<p>– Electrostatic self</p>
<p>– Electrostatic correction</p>
<p>– Total electrostatic energy (sum of real, reciprocal, self, and correction)</p>
<p>– Total Energy (sum of the all energies)</p>
</li>
<li><p class="first">Pressure, Pressure Tensor (<span class="math notranslate nohighlight">\(P_{xx},P_{yy},P_{zz}\)</span>)</p>
</li>
<li><p class="first">Volume</p>
</li>
<li><p class="first">Total molecule number</p>
</li>
<li><p class="first">Total Density</p>
</li>
<li><p class="first">Surface Tension</p>
</li>
<li><p class="first">Mole fraction of each species</p>
</li>
<li><p class="first">Mole density of each species</p>
</li>
</ul>
<p>Detailed move, energy, and statistical or thermodynamic information for each simulation box will be printed in three different sections. Each section’s title will start with <code class="docutils literal notranslate"><span class="pre">MTITLE</span></code>, <code class="docutils literal notranslate"><span class="pre">ETITLE</span></code>, and <code class="docutils literal notranslate"><span class="pre">STITLE</span></code> for move, energy, and statistical information, respectively. The instantaneous values for each section will start with <code class="docutils literal notranslate"><span class="pre">MOVE_#</span></code>, <code class="docutils literal notranslate"><span class="pre">ENER_#</span></code>, and <code class="docutils literal notranslate"><span class="pre">STAT_#</span></code> for move, energy, and statistical values, respectively. Where, # is the simulation box number. In addition, if pressure calculation is activated and enabled to print, pressure tensor will be printed in the console output file. This section starts with <code class="docutils literal notranslate"><span class="pre">PRES_#</span></code> and print the diagonal value of pressure tensor <span class="math notranslate nohighlight">\(P_{xx}\)</span>, <span class="math notranslate nohighlight">\(P_{yy}\)</span>,and <span class="math notranslate nohighlight">\(P_{zz}\)</span>, respectively. The second element after the title of each section is the step number.</p>
<p>In order to extract the desired information from the console file, “grep” and “awk” commands can be used with a proper title section. For example, in order to extract total energy of the system, the following command needs to be executed in terminal:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ grep <span class="s2">&quot;ENER_0&quot;</span> output_console.log <span class="p">|</span> awk <span class="s1">&#39;{print $3}&#39;</span>
</pre></div>
</div>
<p>Here, “output_console.log” is the console output file and “$3” represents the second element of the “ENERGY_BOX_0” section.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p><em>Surface Tension is calculated using Virial method according to following equation</em>,</p>
<div class="last math notranslate nohighlight">
\[\gamma = \frac{1}{2A_{xy}} \int_{0}^{L} \bigg(P_{zz} - \frac{P_{xx} + P_{yy}}{2} \bigg) dz\]</div>
</div>
<p>The first section of this console output typically includes some information relating the system, CPU, GPU, and RAM. In continue, console output includes information regarding the input file (configuration file), force field reading, summary of the structure of the molecule, bonded and non-bonded parameters, and minimum and maximum coordinate of molecules. This output is important; it may contain text relating to issues encountered if there was an error in the current run (e.g. a bad parameter, unknown keyword, missing parameters in the configuration file, etc.)</p>
<img alt="_images/out1.png" src="_images/out1.png" />
<img alt="_images/out2.png" src="_images/out2.png" />
<img alt="_images/out3.png" src="_images/out3.png" />
<p>Next, the energy and statistic title, initial energy and statistic of the system’s starting configuration will print:</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">If total energy of simulation is greater that <span class="math notranslate nohighlight">\(1.0e^{12}\)</span>, System Total Energy Calculation will be performed at EqSteps to preserve energy value.</p>
</div>
<img alt="_images/out4.png" src="_images/out4.png" />
<p>After the simulation starts, move, energy, and statistical title, followed by their values for each simulation box, will print:</p>
<img alt="_images/out5.png" src="_images/out5.png" />
<p>At the end of the run, Monte Carlo move acceptance for each molecule kind and simulation box, total amount of time spent on each Monte Carlo move, total timing information, and other wrap up info will be printed.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>Printed energy and statistical values are instantaneous values.</li>
<li>In order to keep the format of console file consistent and print the calculated properties with high accuracy, scientific format is used.</li>
<li>It’s important to watch the acceptance rates and adjust the move percentages and CBMC trial amounts to get the desired rate of move acceptance.</li>
</ul>
</div>
</div>
<div class="section" id="block-output-files">
<h3>Block Output Files<a class="headerlink" href="#block-output-files" title="Permalink to this headline">¶</a></h3>
<p>GOMC tracks a number of thermodynamic variables of interest during the simulation and prints them all in one file for each box.</p>
<ul>
<li><p class="first">Energy</p>
<p>– Intermolecular (LJ)</p>
<p>– Intramolecular bonded</p>
<p>– Intramolecular nonbonded</p>
<p>– Tail corrections</p>
<p>– Electrostatic real</p>
<p>– Electrostatic Reciprocal</p>
<p>– Total Energy (sum of the all energies)</p>
</li>
<li><p class="first">Virial</p>
</li>
<li><p class="first">Pressure</p>
</li>
<li><p class="first">Surface Tension (using virial method)</p>
</li>
<li><p class="first">Volume</p>
</li>
<li><p class="first">Total molecule number</p>
</li>
<li><p class="first">Total Density</p>
</li>
<li><p class="first">Mole fraction of each species</p>
</li>
<li><p class="first">Mole density of each species</p>
</li>
<li><p class="first">Heat of vaporization</p>
</li>
</ul>
<p>At the beginning of each file, the title of each property followed by their average values is printed. Desired data can be extracted, as explained before, using the “awk” command. For example, in order to extract total density of the system, the following command need to be executed in terminal:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ cat Blk_OutputName_BOX_0.dat <span class="p">|</span> awk <span class="s1">&#39;{print $13}&#39;</span>
</pre></div>
</div>
<p>Here, “Blk_OutputName_BOX_0.dat” is the block-average file for simulation box 0 and “$13” represents the 13th column of the block file.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">In order to keep the format of BlockOutput file consistent and print the calculated properties with high accuracy, scientific format is used.</p>
</div>
</div>
<div class="section" id="visualizing-simulation">
<h3>Visualizing Simulation<a class="headerlink" href="#visualizing-simulation" title="Permalink to this headline">¶</a></h3>
<p>If <code class="docutils literal notranslate"><span class="pre">CoordinatesFreq</span></code> is enabled in configuration file, GOMC will output the molecule coordinates every specified stpes. The PDB and PSF output (merging of atom entries) has already been mentioned/explained in previous sections. To recap: The PDB file’s ATOM entries’ occupancy is used to represent the box the molecule is in for the current frame. All molecules are listed in order in which they were read (i.e. if box 0 has <span class="math notranslate nohighlight">\(1, 2, ..., N1\)</span> molecules and box 1 has <span class="math notranslate nohighlight">\(1, 2, ..., N2\)</span> molecules, then all of the molecules in box 0 are listed first and all the molecules in box 1, i.e. <span class="math notranslate nohighlight">\(1, 2 ,... ,N1\)</span>, <span class="math notranslate nohighlight">\(N1 + 1, ..., N1 + N2\)</span>). PDB frames are written as standard PDBs to consecutive file frames.</p>
<p>To visualize, open the output PDB and PSF files by GOMC using VMD, type this command in the terminal:</p>
<p>For all simulation except Gibbs ensemble that has one simulation box:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ vmd   ISB_T_270_k_merged.psf  ISB_T_270_k_BOX_0.pdb
</pre></div>
</div>
<p>For Gibbs ensemble, visualizing the first box:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ vmd   ISB_T_270_k_merged.psf  ISB_T_270_k_BOX_0.pdb
</pre></div>
</div>
<p>For Gibbs ensemble, visualizing the second box:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ vmd   ISB_T_270_k_merged.psf  ISB_T_270_k_BOX_1.pdb
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Restart coordinate file (OutputName_BOX_0_restart.pdb) cannot be visualize using merged psf file, because atom number does not match. However, you can still open it in vmd using following command and vmd will automatically find the bonds of the molecule based on the coordinates.</p>
</div>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ vmd   ISB_T_270_k_BOX_0_restart.pdb
</pre></div>
</div>
</div>
</div>
<span id="document-putting_all_together"></span><div class="section" id="putting-it-all-together-running-a-gomc-simulation">
<h2>Putting it all together: Running a GOMC Simulation<a class="headerlink" href="#putting-it-all-together-running-a-gomc-simulation" title="Permalink to this headline">¶</a></h2>
<p>It is strongly recommended that you download the test system provided at <a class="reference external" href="http://gomc.eng.wayne.edu/downloads.html">GOMC Website</a> or <a class="reference external" href="https://github.com/GOMC-WSU/GOMC_Examples/tree/master">Our Github Page</a></p>
<p>Run different simulation types in order to become more familiar with different parameter and configuration files (*.conf).</p>
<p>To recap the previous examples, a simulation of isobutane will be completed for a single temperature point on the saturated vapor-liquid coexistence curve.</p>
<p>The general plan for running the simulation is:</p>
<ol class="arabic simple">
<li>Build GOMC (if not done already)</li>
<li>Copy GOMC executable to build directory</li>
<li>Create scripts, PDB, and topology file to build the system, plus in.dat file and parameter files to prepare for runtime</li>
<li>Build finished PDBs and PSFs using the simulation.</li>
<li>Run the simulation in the terminal.</li>
<li>Analyze the output.</li>
</ol>
<p>Please, complete steps 1 and 2; then, traverse to the directory, which should now contain a single file “GOMC_CPU_GEMC”. Next, six files need to be made:</p>
<ul class="simple">
<li>PDB file for isobutane</li>
<li>Topology file describing isobutane residue</li>
<li>Two <code class="docutils literal notranslate"><span class="pre">*.inp</span></code> packmol scripts to pack two system boxes</li>
<li>Two <em>TCL</em> scripts to input into <code class="docutils literal notranslate"><span class="pre">PSFGen</span></code> to generate the final configuration</li>
</ul>
<p><code class="docutils literal notranslate"><span class="pre">isobutane.pdb</span></code></p>
<blockquote>
<div><div class="highlight-text notranslate"><div class="highlight"><pre><span></span>REMARK   1 File  created   by  GaussView   5.0.8
ATOM          1       C1  ISB          1   0.911   -0.313    0.000  C
ATOM          2       C2  ISB          1   1.424   -1.765    0.000  C
ATOM          3       C3  ISB          1  -0.629   -0.313    0.000  C
ATOM          4       C4  ISB          1   1.424    0.413   -1.257  C
END
</pre></div>
</div>
</div></blockquote>
<p><code class="docutils literal notranslate"><span class="pre">Top_Branched_Alkane.inp</span></code></p>
<blockquote>
<div><div class="highlight-text notranslate"><div class="highlight"><pre><span></span>* Custom top file -- branched alkanes
*
MASS     1    CH3      15.035 C !
MASS     2    CH1      13.019 C !

AUTOGENERATE ANGLES DIHEDRALS

RESI   ISB   0.00               !  isobutane { TraPPE }
GROUP
ATOM    C1    CH1       0.00    !  C3\
ATOM    C2    CH3       0.00    !     C2-C1
ATOM    C3    CH3       0.00    !  C4/
ATOM    C4    CH3       0.00    !
BOND    C1  C2   C1  C3   C1  C4
PATCHING FIRS NONE LAST NONE
END
</pre></div>
</div>
</div></blockquote>
<p><code class="docutils literal notranslate"><span class="pre">pack_box_0.inp</span></code></p>
<blockquote>
<div><div class="highlight-text notranslate"><div class="highlight"><pre><span></span>tolerance   3.0
filetype    pdb
output      STEP2_ISB_packed_BOX_0.pdb

structure     isobutane.pdb
number        1000
inside cube   0.  0.  0.  68.00
end structure
</pre></div>
</div>
</div></blockquote>
<p><code class="docutils literal notranslate"><span class="pre">pack_box_1.inp</span></code></p>
<blockquote>
<div><div class="highlight-text notranslate"><div class="highlight"><pre><span></span>tolerance   3.0
filetype    pdb
output      STEP2_ISB_packed_BOX_1.pdb

structure     isobutane.pdb
number        1000
inside cube   0.  0.  0.  68.00
end structure
</pre></div>
</div>
</div></blockquote>
<p><code class="docutils literal notranslate"><span class="pre">build_box_0.inp</span></code></p>
<blockquote>
<div><div class="highlight-text notranslate"><div class="highlight"><pre><span></span>package require psfgen

topology  ./Top Branched Alkane.inp segment ISB {
  pdb     ./STEP2_ISB_packed_BOX_0.pdb
  first   none
  last    none
}
coordpdb  ./STEP2 ISB_packed_BOX_0.pdb ISB

writepsf  ./STEP3_START_ISB_sys_BOX_0.psf
writepdb  ./STEP3_START_ISB_sys_BOX_0.pdb
</pre></div>
</div>
</div></blockquote>
<p><code class="docutils literal notranslate"><span class="pre">build_box_1.inp</span></code></p>
<blockquote>
<div><div class="highlight-text notranslate"><div class="highlight"><pre><span></span>package require psfgen

topology  ./Top Branched Alkane.inp segment ISB {
  pdb     ./STEP2_ISB_packed_BOX_1.pdb
  first   none
  last    none
}
coordpdb  ./STEP2 ISB_packed_BOX_1.pdb ISB

writepsf  ./STEP3_START_ISB_sys_BOX_1.psf
writepdb  ./STEP3_START_ISB_sys_BOX_1.pdb
</pre></div>
</div>
</div></blockquote>
<p>These files can be created with a standard Linux or Windows text editor. Please, also copy a Packmol executable into the working directory.</p>
<p>Once those files are created, run in the terminal:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ ./packmol   &lt;   pack_box_0.inp
$ ./packmol   &lt;   pack_box_1.inp
</pre></div>
</div>
<p>This will create the intermediate PDBs.</p>
<p>Then, run the PSFGen scripts to finish the system using the following commands:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ vmd -dispdev text &lt; ./build_box_0.inp
$ vmd -dispdev text &lt; ./build_box_1.inp
</pre></div>
</div>
<p>This will create the intermediate PDBs.</p>
<p>To run the code a few additional things will be needed:</p>
<ul class="simple">
<li>A GOMC Gibbs ensemble executable</li>
<li>A control file</li>
<li>Parameter files</li>
</ul>
<p>Enter the control file (in.conf) in the text editor in order to modify it. Example files for different simulation types can be found in previous section.</p>
<p>Once these four files have been added to the output directory, the simulation is ready.</p>
<p>Assuming the code is named GOMC_CPU_GEMC, run in the terminal using:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ ./GOMC CPU GEMC in.conf &gt; out_ISB_T_330.00_K_RUN_0.log <span class="p">&amp;</span>
</pre></div>
</div>
<p>For running GOMC in parallel, using openmp, run in the terminal using:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ ./GOMC CPU GEMC +p4 in.conf &gt; out_ISB_T_330.00_K_RUN_0.log<span class="p">&amp;</span>
</pre></div>
</div>
<p>Here, 4 defines the number of processors that will be used to run the simulation in parallel.</p>
<p>Progress can be monitored in the terminal with the tail command:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ tail -f out_ISB.log
</pre></div>
</div>
<div class="admonition attention">
<p class="first admonition-title">Attention</p>
<p class="last">Congratulations! You have examined a single-phase coexistence point on the saturated vapor-liquid curve using GOMC operating in the Gibbs ensemble.</p>
</div>
<div class="figure" id="id1">
<img alt="_images/isobutane_result.png" src="_images/isobutane_result.png" />
<p class="caption"><span class="caption-text">Repeating this process for multiple temperatures will allow you to obtain the following results.</span></p>
</div>
</div>
<span id="document-vdw_energy"></span><div class="section" id="intermolecular-energy-and-virial-function-van-der-waals">
<h2>Intermolecular Energy and Virial Function (Van der Waals)<a class="headerlink" href="#intermolecular-energy-and-virial-function-van-der-waals" title="Permalink to this headline">¶</a></h2>
<p>In this section, the virial and energy equation of Van der Waals interaction for different potential function are discussed in details.</p>
<div class="section" id="vdw">
<h3>VDW<a class="headerlink" href="#vdw" title="Permalink to this headline">¶</a></h3>
<p>This option calculates potential energy without any truncation.</p>
<dl class="docutils">
<dt><code class="docutils literal notranslate"><span class="pre">Potential</span> <span class="pre">Calculation</span></code></dt>
<dd><p class="first">Interactions between atoms can be modeled with an n−6 potential, a Mie potential in which the attractive exponent is fixed. The Mie potential can be viewed as a generalized version of the 12-6 Lennard-Jones potential,</p>
<div class="math notranslate nohighlight">
\[E_{ij} = C_{n_{ij}} \epsilon_{ij} \bigg[\bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^{n_{ij}} - \bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^6\bigg]\]</div>
<p>where <span class="math notranslate nohighlight">\(r_{ij}\)</span>, <span class="math notranslate nohighlight">\(\epsilon_{ij}\)</span>, and <span class="math notranslate nohighlight">\(\sigma_{ij}\)</span> are, respectively, the separation, well depth, and collision diameter for the pair of interaction sites <span class="math notranslate nohighlight">\(i\)</span> and <span class="math notranslate nohighlight">\(j\)</span>. The constant <span class="math notranslate nohighlight">\(C_n\)</span> is a normalization factor such that the minimum of the potential remains at <span class="math notranslate nohighlight">\(−\epsilon_{ij}\)</span> for all <span class="math notranslate nohighlight">\(n_{ij}\)</span>. In the 12-6 potential, <span class="math notranslate nohighlight">\(C_n\)</span> reduces to the familiar value of 4.</p>
<div class="last math notranslate nohighlight">
\[C_{n_{ij}} = \bigg(\frac{n_{ij}}{n_{ij} - 6} \bigg)\bigg(\frac{n_{ij}}{6} \bigg)^{6/(n_{ij} - 6)}\]</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">Virial</span> <span class="pre">Calculation</span></code></dt>
<dd><p class="first">Virial is basically the negative derivative of energy with respect to distance, multiplied by distance.</p>
<div class="math notranslate nohighlight">
\[W_{ij} = -\frac{dE_{ij}}{dr}\times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}}\]</div>
<p>Using n−6 LJ potential defined above:</p>
<div class="last math notranslate nohighlight">
\[W_{ij} = 6C_{n_{ij}} \epsilon_{ij} \bigg[\frac{n_{ij}}{6} \times \bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^{n_{ij}} - \bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^6\bigg]\times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}^2}\]</div>
</dd>
</dl>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This option only evaluates the energy up to specified <code class="docutils literal notranslate"><span class="pre">Rcut</span></code> distance. Tail correction to energy and pressure can be specified to account for infinite cutoff distance.</p>
</div>
</div>
<div class="section" id="shift">
<h3>SHIFT<a class="headerlink" href="#shift" title="Permalink to this headline">¶</a></h3>
<p>This option forces the potential energy to be zero at <code class="docutils literal notranslate"><span class="pre">Rcut</span></code> distance.</p>
<dl class="docutils">
<dt><code class="docutils literal notranslate"><span class="pre">Potential</span> <span class="pre">Calculation</span></code></dt>
<dd><p class="first">Interactions between atoms can be modeled with an n−6 potential,</p>
<div class="math notranslate nohighlight">
\[E_{ij}(\texttt{shift}) = C_{n_{ij}} \epsilon_{ij} \bigg[\bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^{n_{ij}} - \bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^6\bigg] - C_{n_{ij}} \epsilon_{ij} \bigg[\bigg(\frac{\sigma_{ij}}{r_{cut}}\bigg)^{n_{ij}} - \bigg(\frac{\sigma_{ij}}{r_{cut}}\bigg)^6\bigg]\]</div>
<p class="last">where <span class="math notranslate nohighlight">\(r_{ij}\)</span>, <span class="math notranslate nohighlight">\(\epsilon_{ij}\)</span>, and <span class="math notranslate nohighlight">\(\sigma_{ij}\)</span> are, respectively, the separation, well depth, and collision diameter for the pair of interaction sites <span class="math notranslate nohighlight">\(i\)</span> and <span class="math notranslate nohighlight">\(j\)</span>. The constant <span class="math notranslate nohighlight">\(C_n\)</span> is a normalization factor according to Eq. 3, such that the minimum of the potential remains at <span class="math notranslate nohighlight">\(−\epsilon_{ij}\)</span> for all <span class="math notranslate nohighlight">\(n_{ij}\)</span>. In the 12-6 potential, <span class="math notranslate nohighlight">\(C_n\)</span> reduces to the familiar value of 4.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">Virial</span> <span class="pre">Calculation</span></code></dt>
<dd><p class="first">Virial is basically the negative derivative of energy with respect to distance, multiplied by distance, Eq. 4.</p>
<p>Using <code class="docutils literal notranslate"><span class="pre">SHIFT</span></code> potential function defined above:</p>
<div class="math notranslate nohighlight">
\[W_{ij}(\texttt{shift}) = 6C_{n_{ij}} \epsilon_{ij} \bigg[\frac{n_{ij}}{6} \times \bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^{n_{ij}} - \bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^6\bigg]\times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}^2}\]</div>
<div class="last figure" id="id1">
<img alt="_images/VDW_SHIFT.png" src="_images/VDW_SHIFT.png" />
<p class="caption"><span class="caption-text">Graph of Van der Waals potential with and without the application of the <code class="docutils literal notranslate"><span class="pre">SHIFT</span></code> function. With the <code class="docutils literal notranslate"><span class="pre">SHIFT</span></code> function active, the potential by force was reduced to 0.0 at the <code class="docutils literal notranslate"><span class="pre">Rcut</span></code> distance. With the <code class="docutils literal notranslate"><span class="pre">SHIFT</span></code> function, there is a discontinuity where the potential is truncated.</span></p>
</div>
</dd>
</dl>
</div>
<div class="section" id="switch">
<h3>SWITCH<a class="headerlink" href="#switch" title="Permalink to this headline">¶</a></h3>
<p>This option in <code class="docutils literal notranslate"><span class="pre">CHARMM</span></code> or <code class="docutils literal notranslate"><span class="pre">EXOTIC</span></code> force field smoothly forces the potential energy to be zero at Rcut distance and starts modifying the potential at Rswitch distance.</p>
<dl class="docutils">
<dt><code class="docutils literal notranslate"><span class="pre">Potential</span> <span class="pre">Calculation</span></code></dt>
<dd><p class="first">Interactions between atoms can be modeled with an n−6 potential,</p>
<div class="math notranslate nohighlight">
\[E_{ij}(\texttt{switch}) = C_{n_{ij}} \epsilon_{ij} \bigg[\bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^{n_{ij}} - \bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^6\bigg]\times F_E\]</div>
<p>where <span class="math notranslate nohighlight">\(r_{ij}\)</span>, <span class="math notranslate nohighlight">\(\epsilon_{ij}\)</span>, and <span class="math notranslate nohighlight">\(\sigma_{ij}\)</span> are, respectively, the separation, well depth, and collision diameter for the pair of interaction sites <span class="math notranslate nohighlight">\(i\)</span> and <span class="math notranslate nohighlight">\(j\)</span>. The constant <span class="math notranslate nohighlight">\(C_n\)</span> is a normalization factor according to Eq. 3, such that the minimum of the potential remains at <span class="math notranslate nohighlight">\(−\epsilon_{ij}\)</span> for all <span class="math notranslate nohighlight">\(n_{ij}\)</span>. In the 12-6 potential, <span class="math notranslate nohighlight">\(C_n\)</span> reduces to the familiar value of 4.</p>
<p>The factor <span class="math notranslate nohighlight">\(F_E\)</span> is defined as:</p>
<div class="last math notranslate nohighlight">
\[\begin{split}F_E =
\begin{cases}
  1 &amp; r_{ij} \leq r_{switch} \\
  \frac{\big({r_{cut}}^2 - {r_{ij}}^2 \big)^2 \times \big({r_{cut}}^2 - 3{r_{switch}}^2 + 2{r_{ij}}^2 \big)}{\big({r_{cut}}^2 - {r_{switch}}^2 \big)^3} &amp; r_{switch} &lt; r_{ij} &lt; r_{cut} \\
  0 &amp; r_{ij} \geq r_{cut}
\end{cases}\end{split}\]</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">Virial</span> <span class="pre">Calculation</span></code></dt>
<dd><p class="first">Virial is basically the negative derivative of energy with respect to distance, multiplied by distance, Eq. 4.</p>
<p>Using SWITCH potential function defined above:</p>
<div class="math notranslate nohighlight">
\[ \begin{align}\begin{aligned}W_{ij}(\texttt{switch}) = \Bigg[6 C_{n_{ij}} \epsilon_{ij} \bigg[\frac{n_{ij}}{6} \times \bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^{n_{ij}} - \bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^6\bigg]\times \frac{F_E}{{r_{ij}}^2}  -\\C_{n_{ij}} \epsilon_{ij} \bigg[\bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^{n_{ij}} - \bigg(\frac{\sigma_{ij}}{r_{ij}}\bigg)^6\bigg] \times F_W \Bigg] \times \overrightarrow{r_{ij}}\end{aligned}\end{align} \]</div>
<p>The factor <span class="math notranslate nohighlight">\(F_W\)</span> is defined as:</p>
<div class="math notranslate nohighlight">
\[\begin{split}F_W =
\begin{cases}
  1 &amp; r_{ij} \leq r_{switch} \\
  \frac{12\big({r_{cut}}^2 - {r_{ij}}^2 \big) \times \big({r_{switch}}^2 - {r_{ij}}^2 \big)}{\big({r_{cut}}^2 - {r_{switch}}^2 \big)^3} &amp; r_{switch} &lt; r_{ij} &lt; r_{cut} \\
  0 &amp; r_{ij} \geq r_{cut}
\end{cases}\end{split}\]</div>
<div class="last figure" id="id2">
<img alt="_images/SWITCH.png" src="_images/SWITCH.png" />
<p class="caption"><span class="caption-text">Graph of Van der Waals potential with and without the application of the <code class="docutils literal notranslate"><span class="pre">SWITCH</span></code> function. With the <code class="docutils literal notranslate"><span class="pre">SWITCH</span></code> function active, the potential is smoothly reduced to 0.0 at the <code class="docutils literal notranslate"><span class="pre">Rcut</span></code> distance.</span></p>
</div>
</dd>
</dl>
</div>
<div class="section" id="switch-martini">
<h3>SWITCH (MARTINI)<a class="headerlink" href="#switch-martini" title="Permalink to this headline">¶</a></h3>
<p>This option in <code class="docutils literal notranslate"><span class="pre">MARTINI</span></code> force field smoothly forces the potential energy to be zero at Rcut distance and starts modifying the potential at <code class="docutils literal notranslate"><span class="pre">Rswitch</span></code> distance.</p>
<dl class="docutils">
<dt><code class="docutils literal notranslate"><span class="pre">Potential</span> <span class="pre">Calculation</span></code></dt>
<dd><p class="first">Potential Calculation: Interactions between atoms can be modeled with an n−6 potential. In standard MARTINI, <span class="math notranslate nohighlight">\(n\)</span> is equal to 12,</p>
<div class="math notranslate nohighlight">
\[E_{ij}(\texttt{switch}) = C_{n_{ij}}\epsilon_{ij} \Bigg[ {\sigma_{ij}}^{n} \bigg(\frac{1}{{r_{ij}}^{n}} + \varphi_{n} (r_{ij}) \bigg) - {\sigma_{ij}}^{6} \bigg(\frac{1}{{r_{ij}}^{6}} + \varphi_{6} (r_{ij}) \bigg) \Bigg]\]</div>
<p>where <span class="math notranslate nohighlight">\(r_{ij}\)</span>, <span class="math notranslate nohighlight">\(\epsilon_{ij}\)</span>, and <span class="math notranslate nohighlight">\(\sigma_{ij}\)</span> are, respectively, the separation, well depth, and collision diameter for the pair of interaction sites <span class="math notranslate nohighlight">\(i\)</span> and <span class="math notranslate nohighlight">\(j\)</span>. The constant <span class="math notranslate nohighlight">\(C_n\)</span> is a normalization factor according to Eq. 3, such that the minimum of the potential remains at <span class="math notranslate nohighlight">\(−\epsilon_{ij}\)</span> for all <span class="math notranslate nohighlight">\(n_{ij}\)</span>. In the 12-6 potential, <span class="math notranslate nohighlight">\(C_n\)</span> reduces to the familiar value of 4.</p>
<p>The factor <span class="math notranslate nohighlight">\(\varphi_{\alpha}\)</span> and constants are defined as:</p>
<div class="math notranslate nohighlight">
\[\begin{split}\varphi_{\alpha}(r_{ij}) =
\begin{cases}
  -C_{\alpha} &amp; r_{ij} \leq r_{switch} \\
  -\frac{A_{\alpha}}{3} (r_{ij} - r_{switch})^3 -\frac{B_{\alpha}}{4} (r_{ij} - r_{switch})^4 - C_{\alpha} &amp; r_{switch} &lt; r_{ij} &lt; r_{cut} \\
  0 &amp; r_{ij} \geq r_{cut}
\end{cases}\end{split}\]</div>
<div class="math notranslate nohighlight">
\[A_{\alpha} = \alpha \frac{(\alpha + 1) r_{switch} - (\alpha +4) r_{cut}} {{r_{cut}}^{(\alpha + 2)} {(r_{cut} - r_{switch})}^2}\]</div>
<div class="math notranslate nohighlight">
\[B_{\alpha} = \alpha \frac{(\alpha + 1) r_{switch} - (\alpha +3) r_{cut}} {{r_{cut}}^{(\alpha + 2)} {(r_{cut} - r_{switch})}^3}\]</div>
<div class="last math notranslate nohighlight">
\[C_{\alpha} =  \frac{1}{{r_{cut}}^{\alpha}} -\frac{A_{\alpha}}{3} (r_{cut} - r_{switch})^3 -\frac{B_{\alpha}}{4} (r_{cut} - r_{switch})^4\]</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">Virial</span> <span class="pre">Calculation</span></code></dt>
<dd><p class="first">Virial is basically the negative derivative of energy with respect to distance, mul- tiplied by distance, Eq. 4.</p>
<p>Using the <code class="docutils literal notranslate"><span class="pre">SWITCH</span></code> potential function defined for <code class="docutils literal notranslate"><span class="pre">MARTINI</span></code> force field:</p>
<div class="math notranslate nohighlight">
\[W_{ij}(\texttt{switch}) = C_{n_{ij}}\epsilon_{ij} \Bigg[ {\sigma_{ij}}^{n} \bigg(\frac{n}{{r_{ij}}^{(n+1)}} + d\varphi_{n} (r_{ij}) \bigg) - {\sigma_{ij}}^{6} \bigg(\frac{6}{{r_{ij}}^{(6+1)}} +d \varphi_{6} (r_{ij}) \bigg) \Bigg]\times \frac{\overrightarrow{r_{ij}}}{r_{ij}}\]</div>
<p>The constants defined in Eq. 14-16 and the factor <span class="math notranslate nohighlight">\(d\varphi_{\alpha}\)</span> defined as:</p>
<div class="math notranslate nohighlight">
\[\begin{split}d\varphi_{\alpha}(r_{ij}) =
\begin{cases}
  0 &amp; r_{ij} \leq r_{switch} \\
  A_{\alpha} (r_{ij} - r_{switch})^2 + B_{\alpha} (r_{ij} - r_{switch})^3 &amp; r_{switch} &lt; r_{ij} &lt; r_{cut} \\
  0 &amp; r_{ij} \geq r_{cut}
\end{cases}\end{split}\]</div>
<div class="last figure" id="id3">
<img alt="_images/MARTINI.png" src="_images/MARTINI.png" />
<p class="caption"><span class="caption-text">Graph of Van der Waals potential with and without the application of the <code class="docutils literal notranslate"><span class="pre">SWITCH</span></code> function in <code class="docutils literal notranslate"><span class="pre">MARTINI</span></code> force field. With the <code class="docutils literal notranslate"><span class="pre">SWITCH</span></code> function active, the potential is smoothly reduced to 0.0 at the <code class="docutils literal notranslate"><span class="pre">Rcut</span></code> distance.</span></p>
</div>
</dd>
</dl>
</div>
</div>
<span id="document-electrostatic"></span><div class="section" id="intermolecular-energy-and-virial-function-electrostatic">
<h2>Intermolecular Energy and Virial Function (Electrostatic)<a class="headerlink" href="#intermolecular-energy-and-virial-function-electrostatic" title="Permalink to this headline">¶</a></h2>
<p>In this section, the virial and energy equation of electrostatic interaction for different potential function are discussed in details.</p>
<div class="section" id="ewald">
<h3>Ewald<a class="headerlink" href="#ewald" title="Permalink to this headline">¶</a></h3>
<p>This option calculate electrostatic energy using standard <em>Ewald Summation Method</em>.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Once this option is activated, it would override the the electrostatic calculation using <code class="docutils literal notranslate"><span class="pre">VDW</span></code>, <code class="docutils literal notranslate"><span class="pre">SHIFT</span></code>, and <code class="docutils literal notranslate"><span class="pre">SWITCH</span></code> functions.</p>
</div>
<dl class="docutils">
<dt><code class="docutils literal notranslate"><span class="pre">Potential</span> <span class="pre">Calculation</span></code></dt>
<dd><p class="first">Coulomb interactions between atoms can be modeled as</p>
<div class="math notranslate nohighlight">
\[E(\texttt{Ewald}) = E_{real} + E_{reciprocal} + E_{self} + E_{correction}\]</div>
<p><span class="math notranslate nohighlight">\(E_{real}\)</span>: Defines the short range electrostatic energy according to</p>
<div class="math notranslate nohighlight">
\[E_{real} = \frac{1}{4\pi \epsilon_0} \frac{1}{2} \sum_{i =1}^{N} \sum_{j = 1}^{N} q_i q_j  \frac{erfc(\alpha r_{ij})}{r_{ij}}\]</div>
<p>, where <span class="math notranslate nohighlight">\(\alpha\)</span> is <code class="docutils literal notranslate"><span class="pre">Ewald</span></code> separation parameter according to</p>
<div class="math notranslate nohighlight">
\[\alpha = \frac {\sqrt{-\log (Tolerance)}}{r_{cut}}\]</div>
<p>, where <code class="docutils literal notranslate"><span class="pre">Tolerance</span></code> is a parameter, controlling the desired accuracy.</p>
<p><span class="math notranslate nohighlight">\(E_{reciprocal}\)</span>: Defines the long range electrostatic energy according to,</p>
<div class="math notranslate nohighlight">
\[E_{reciprocal} = \frac{1}{\epsilon_0 V} \frac {1}{2} \sum_{\overrightarrow{k} \ne 0}^{} \frac {1}{\overrightarrow{k}^2}\exp\bigg(\frac {-\overrightarrow{k}^2}{4 \alpha^2}\bigg) \Bigg[ {\Big| R_{sum} \Big|}^2 + {\Big| I_{sum} \Big|}^2 \bigg]\]</div>
<p>, where <span class="math notranslate nohighlight">\(\overrightarrow{k}\)</span> is reciprocal vector, <span class="math notranslate nohighlight">\(R_{sum}\)</span> and <span class="math notranslate nohighlight">\(I_{sum}\)</span> are,</p>
<div class="math notranslate nohighlight">
\[R_{sum} = \sum_{i=1}^{N} q_i \cos \big(\overrightarrow{k}.\overrightarrow{x_i}\big)\]</div>
<div class="math notranslate nohighlight">
\[I_{sum} = \sum_{i=1}^{N} q_i \sin \big(\overrightarrow{k}.\overrightarrow{x_i}\big)\]</div>
<p><span class="math notranslate nohighlight">\(E_{self}\)</span>: Defines the self energy according to,</p>
<div class="math notranslate nohighlight">
\[E_{self} = -\frac{\alpha}{4\pi \epsilon_0 \sqrt{\pi}} \sum_{i=1}^{N} {q_i}^2\]</div>
<p><span class="math notranslate nohighlight">\(E_{correction}\)</span>: Defines intra-molecule nonbonded energy,</p>
<div class="last math notranslate nohighlight">
\[E_{correction} = -\frac{1}{4\pi \epsilon_0} \frac{1}{2} \sum_{j=1}^{N }\sum_{l =1}^{N_j} \sum_{m = 1}^{N_j} q_{j_l} q_{j_m}  \frac{erf(\alpha r_{j_l j_m})}{r_{j_l j_m}}\]</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">Virial</span> <span class="pre">Calculation</span></code></dt>
<dd><p class="first">Virial is basically the negative derivative of energy with respect to distance, multiplied by distance, Eq. 4. Coulomb force between atoms can be modeled as,</p>
<div class="math notranslate nohighlight">
\[W_{Ewald} = W_{real} + W_{reciprocal}\]</div>
<p><span class="math notranslate nohighlight">\(W_{real}\)</span> defines the short range electrostatic and <span class="math notranslate nohighlight">\(W_{reciprocal}\)</span> defines the long range electrostatic force according to,</p>
<div class="math notranslate nohighlight">
\[W_{real} = \frac{1}{4\pi \epsilon_0} \frac{1}{2} \sum_{i =1}^{N} \sum_{j = 1}^{N} q_i q_j  \bigg[ \frac{erfc(\alpha r_{ij})}{r_{ij}} + \frac{2\alpha}{ \sqrt{\pi}} \exp(-\alpha^2 {r_{ij}}^2) \bigg] \times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}^2}\]</div>
<div class="math notranslate nohighlight">
\[\begin{split}\begin{split}
              W_{reciprocal} = \frac{1}{\epsilon_0 V} \frac {1}{2} \sum_{\overrightarrow{k} \ne 0}^{} \Bigg[\frac {1}{\overrightarrow{k}^2}\exp\bigg(\frac {-\overrightarrow{k}^2}{4 \alpha^2}\bigg) \bigg( {\Big| R_{sum} \Big|}^2 + {\Big| I_{sum} \Big|}^2 \bigg) \bigg(  1 - \frac{\overrightarrow{k}^2}{2\alpha^2} \bigg) \Bigg] +\\
  \sum_{i=1}^{N} \frac{1}{\epsilon_0 V}  \sum_{\overrightarrow{k} \ne 0}^{} \Bigg[ \frac {q_i}{\overrightarrow{k}^2}\exp\bigg(\frac {-\overrightarrow{k}^2}{4 \alpha^2}\bigg) \bigg[ I_{sum} \times\cos(\overrightarrow{k}.\overrightarrow{x_i})  - R_{sum} \times \sin(\overrightarrow{k}.\overrightarrow{x_i}) \bigg] \Bigg] \times \big( \overrightarrow{k}.\overrightarrow{r_{ic}} \big)
      \end{split}\end{split}\]</div>
<p class="last">, where <span class="math notranslate nohighlight">\(\overrightarrow{r_{ic}}\)</span> is the vector between atom and the center of the mass of the molecule.</p>
</dd>
</dl>
</div>
<div class="section" id="shift">
<h3>SHIFT<a class="headerlink" href="#shift" title="Permalink to this headline">¶</a></h3>
<p>This option forces the electrostatic energy to be zero at <code class="docutils literal notranslate"><span class="pre">Rcut</span></code> distance.</p>
<dl class="docutils">
<dt><code class="docutils literal notranslate"><span class="pre">Potential</span> <span class="pre">Calculation</span></code></dt>
<dd><p class="first">Coulomb interactions between atoms can be modeled as</p>
<div class="last math notranslate nohighlight">
\[E(\texttt{SHIFT}) = \frac{q_i q_j}{4\pi \epsilon_0} \Big( \frac{1}{r_{ij}} - \frac{1}{r_{cut}} \Big)\]</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">Virial</span> <span class="pre">Calculation</span></code></dt>
<dd><p class="first">Virial is basically the negative derivative of energy with respect to distance, multiplied by distance, Eq. 4. Coulomb force between atoms can be modeled as,</p>
<blockquote class="last">
<div><div class="math notranslate nohighlight">
\[W(\texttt{SHIFT}) = \frac{q_i q_j}{4\pi \epsilon_0} \Big( \frac{1}{r_{ij}} \times \frac{\overrightarrow{r_{ij}}}{{r_{ij}}^2} \Big)\]</div>
</div></blockquote>
</dd>
</dl>
</div>
<div class="section" id="switch">
<h3>SWITCH<a class="headerlink" href="#switch" title="Permalink to this headline">¶</a></h3>
<p>This option in <code class="docutils literal notranslate"><span class="pre">CHARMM</span></code> or <code class="docutils literal notranslate"><span class="pre">EXOTIC</span></code> force field forces the electrostatic energy to be zero at <code class="docutils literal notranslate"><span class="pre">Rcut</span></code> distance.</p>
<dl class="docutils">
<dt><code class="docutils literal notranslate"><span class="pre">Potential</span> <span class="pre">Calculation</span></code></dt>
<dd><p class="first">Coulomb interactions between atoms can be modeled as,</p>
<div class="last math notranslate nohighlight">
\[E(\texttt{SWITCH}) = \frac{q_i q_j}{4\pi \epsilon_0} \bigg( \Big(\frac{r_{ij}}{r_{cut}} \Big)^2 - 1.0\bigg)^2 \frac{1}{r_{ij}}\]</div>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">Virial</span> <span class="pre">Calculation</span></code></dt>
<dd><p class="first">Virial is basically the negative derivative of energy with respect to distance, multiplied by distance, Eq. 4. Coulomb force between atoms can be modeled as,</p>
<div class="last math notranslate nohighlight">
\[W(\texttt{SWITCH}) = \frac{q_i q_j}{4\pi \epsilon_0} \Bigg[ \bigg( \Big(\frac{r_{ij}}{r_{cut}} \Big)^2 - 1.0\bigg)^2 \frac{1}{{r_{ij}}^2} - \bigg( \frac{4}{{r_{cut}}^2} \bigg) \bigg( \Big(\frac{r_{ij}}{r_{cut}} \Big)^2 - 1.0\bigg) \Bigg] \times \frac{\overrightarrow{r_{ij}}}{r_{ij}}\]</div>
</dd>
</dl>
</div>
<div class="section" id="switch-martini">
<h3>SWITCH (MARTINI)<a class="headerlink" href="#switch-martini" title="Permalink to this headline">¶</a></h3>
<p>This option in <code class="docutils literal notranslate"><span class="pre">MARTINI</span></code> force field smoothly forces the potential energy to be zero at <code class="docutils literal notranslate"><span class="pre">Rcut</span></code> distance and starts modifying the potential at <code class="docutils literal notranslate"><span class="pre">Rswitch</span> <span class="pre">=</span> <span class="pre">0.0</span></code> distance.</p>
<dl class="docutils">
<dt><code class="docutils literal notranslate"><span class="pre">Potential</span> <span class="pre">Calculation</span></code></dt>
<dd><p class="first">Coulomb interactions between atoms can be modeled as,</p>
<div class="math notranslate nohighlight">
\[E(\texttt{SWITCH})=\frac{q_i q_j}{4\pi\epsilon_0\epsilon_1}\bigg(\frac{1}{r_{ij}}+\varphi_{1}(r_{ij})\bigg)\]</div>
<p class="last">, where <span class="math notranslate nohighlight">\(\epsilon_1\)</span> is the dielectric constant, which in <code class="docutils literal notranslate"><span class="pre">MARTINI</span></code> force field is equal to 15.0 and <span class="math notranslate nohighlight">\(\varphi_{\alpha}(r_{ij})\)</span> is defined in Eq. 13-16.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">Virial</span> <span class="pre">Calculation</span></code></dt>
<dd><p class="first">Virial is basically the negative derivative of energy with respect to distance, multiplied by distance, Eq. 4. Coulomb force between atoms can be modeled as,</p>
<div class="math notranslate nohighlight">
\[W(\texttt{SWITCH})=\frac{q_iq_j}{4\pi\epsilon_0\epsilon_1}\bigg(\frac{1}{{r_{ij}}^2}+d\varphi_1(r_{ij})\bigg)\times\frac{\overrightarrow{r_{ij}}}{r_{ij}}\]</div>
<p class="last">, where <span class="math notranslate nohighlight">\(d\varphi_1 (r_{ij})\)</span> is defined in Eq. 18.</p>
</dd>
</dl>
</div>
</div>
<span id="document-help"></span><div class="section" id="get-help-or-technical-support">
<h2>Get Help or Technical Support<a class="headerlink" href="#get-help-or-technical-support" title="Permalink to this headline">¶</a></h2>
<p>For get any help or technical support, please send message to GOMC gitter:</p>
<p><a class="reference external" href="https://gitter.im/GOMC_WSU/Lobby">https://gitter.im/GOMC_WSU/Lobby</a></p>
<p>or send email to:</p>
<ul class="simple">
<li>Jeffrey Potoff: <a class="reference external" href="mailto:jpotoff&#37;&#52;&#48;wayne&#46;edu">jpotoff<span>&#64;</span>wayne<span>&#46;</span>edu</a></li>
<li>Loren Schwiebert: <a class="reference external" href="mailto:loren&#37;&#52;&#48;wayne&#46;edu">loren<span>&#64;</span>wayne<span>&#46;</span>edu</a></li>
</ul>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019, GOMC

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>